gnomAD - Population Specific variants of all genes - Population unique variants extracted & Tables of frequencies of non-annotated and pathogenic variants and Distribution of population specific variants among clinical significance and gene.

```{r}
library(tidyverse)
library(here)
library(janitor)
library(visdat)
library(skimr)
library(rstatix)
library(knitr)
```

Checking for unique variants
ATM
African
```{r}
atm_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> atm_afr_ben

atm_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> atm_afr_path

atm_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> atm_afr_uncertain

atm_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> atm_afr_blank
```
European
```{r}
atm_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> atm_euro_ben

atm_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> atm_euro_path

atm_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> atm_euro_uncertain

atm_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> atm_euro_blank
```
Latino
```{r}
atm_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> atm_lat_ben

atm_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> atm_lat_path

atm_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> atm_lat_uncertain

atm_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> atm_lat_blank
```
East Asia
```{r}
atm_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> atm_ea_ben

atm_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> atm_ea_path

atm_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> atm_ea_uncertain

atm_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> atm_ea_blank
```
South Asia
```{r}
atm_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> atm_sa_ben

atm_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> atm_sa_path

atm_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> atm_sa_uncertain

atm_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> atm_sa_blank
```

BARD1
African
```{r}
bard1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> bard1_afr_ben

bard1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> bard1_afr_path

bard1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> bard1_afr_uncertain

bard1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> bard1_afr_blank
```
European
```{r}
bard1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> bard1_euro_ben

bard1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> bard1_euro_path

bard1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> bard1_euro_uncertain

bard1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> bard1_euro_blank
```
Latino
```{r}
bard1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> bard1_lat_ben

bard1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> bard1_lat_path

bard1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> bard1_lat_uncertain

bard1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> bard1_lat_blank
```
East Asia
```{r}
bard1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> bard1_ea_ben

bard1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> bard1_ea_path

bard1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> bard1_ea_uncertain

bard1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> bard1_ea_blank
```
South Asia
```{r}
bard1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> bard1_sa_ben

bard1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> bard1_sa_path

bard1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> bard1_sa_uncertain

bard1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> bard1_sa_blank
```

BRCA1
African
```{r}
brca1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca1_afr_ben

brca1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca1_afr_path

brca1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca1_afr_uncertain

brca1_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca1_afr_blank
```
European
```{r}
brca1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca1_euro_ben

brca1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca1_euro_path

brca1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca1_euro_uncertain

brca1_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca1_euro_blank
```
Latino
```{r}
brca1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca1_lat_ben

brca1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca1_lat_path

brca1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca1_lat_uncertain

brca1_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca1_lat_blank
```
East Asia
```{r}
brca1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca1_ea_ben

brca1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca1_ea_path

brca1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca1_ea_uncertain

brca1_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca1_ea_blank
```
South Asia
```{r}
brca1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> brca1_sa_ben

brca1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> brca1_sa_path

brca1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> brca1_sa_uncertain

brca1_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> brca1_sa_blank
```

BRCA2
African
```{r}
brca2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca2_afr_ben

brca2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca2_afr_path

brca2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca2_afr_uncertain

brca2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca2_afr_blank
```
European
```{r}
brca2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca2_euro_ben

brca2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca2_euro_path

brca2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca2_euro_uncertain

brca2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca2_euro_blank
```
Latino
```{r}
brca2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca2_lat_ben

brca2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca2_lat_path

brca2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca2_lat_uncertain

brca2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca2_lat_blank
```
East Asia
```{r}
brca2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> brca2_ea_ben

brca2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> brca2_ea_path

brca2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> brca2_ea_uncertain

brca2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> brca2_ea_blank
```
South Asia
```{r}
brca2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> brca2_sa_ben

brca2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> brca2_sa_path

brca2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> brca2_sa_uncertain

brca2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> brca2_sa_blank
```

CHEK2
African
```{r}
chek2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> chek2_afr_ben

chek2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> chek2_afr_path

chek2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> chek2_afr_uncertain

chek2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> chek2_afr_blank
```
European
```{r}
chek2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> chek2_euro_ben

chek2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> chek2_euro_path

chek2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> chek2_euro_uncertain

chek2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> chek2_euro_blank
```
Latino
```{r}
chek2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> chek2_lat_ben

chek2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> chek2_lat_path

chek2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> chek2_lat_uncertain

chek2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> chek2_lat_blank
```
East Asia
```{r}
chek2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> chek2_ea_ben

chek2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> chek2_ea_path

chek2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> chek2_ea_uncertain

chek2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> chek2_ea_blank
```
South Asia
```{r}
chek2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> chek2_sa_ben

chek2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> chek2_sa_path

chek2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> chek2_sa_uncertain

chek2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> chek2_sa_blank
```

PALB2
African
```{r}
palb2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> palb2_afr_ben

palb2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> palb2_afr_path

palb2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> palb2_afr_uncertain

palb2_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> palb2_afr_blank
```
European
```{r}
palb2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> palb2_euro_ben

palb2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> palb2_euro_path

palb2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> palb2_euro_uncertain

palb2_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> palb2_euro_blank
```
Latino
```{r}
palb2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> palb2_lat_ben

palb2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> palb2_lat_path

palb2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> palb2_lat_uncertain

palb2_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> palb2_lat_blank
```
East Asia
```{r}
palb2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> palb2_ea_ben

palb2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> palb2_ea_path

palb2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> palb2_ea_uncertain

palb2_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> palb2_ea_blank
```
South Asia
```{r}
palb2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> palb2_sa_ben

palb2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> palb2_sa_path

palb2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> palb2_sa_uncertain

palb2_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> palb2_sa_blank
```

RAD51C
African
```{r}
rad51c_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51c_afr_ben

rad51c_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51c_afr_path

rad51c_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51c_afr_uncertain

rad51c_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51c_afr_blank
```
European
```{r}
rad51c_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51c_euro_ben

rad51c_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51c_euro_path

rad51c_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51c_euro_uncertain

rad51c_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51c_euro_blank
```
Latino
```{r}
rad51c_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51c_lat_ben

rad51c_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51c_lat_path

rad51c_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51c_lat_uncertain

rad51c_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51c_lat_blank
```
East Asia
```{r}
rad51c_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51c_ea_ben

rad51c_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51c_ea_path

rad51c_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51c_ea_uncertain

rad51c_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51c_ea_blank
```
South Asia
```{r}
rad51c_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> rad51c_sa_ben

rad51c_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> rad51c_sa_path

rad51c_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> rad51c_sa_uncertain

rad51c_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> rad51c_sa_blank
```

RAD51D
African
```{r}
rad51d_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51d_afr_ben

rad51d_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51d_afr_path

rad51d_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51d_afr_uncertain

rad51d_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51d_afr_blank
```
European
```{r}
rad51d_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51d_euro_ben

rad51d_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51d_euro_path

rad51d_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51d_euro_uncertain

rad51d_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51d_euro_blank
```
Latino
```{r}
rad51d_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51d_lat_ben

rad51d_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51d_lat_path

rad51d_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51d_lat_uncertain

rad51d_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51d_lat_blank
```
East Asia
```{r}
rad51d_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> rad51d_ea_ben

rad51d_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> rad51d_ea_path

rad51d_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> rad51d_ea_uncertain

rad51d_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> rad51d_ea_blank
```
South Asia
```{r}
rad51d_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> rad51d_sa_ben

rad51d_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> rad51d_sa_path

rad51d_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> rad51d_sa_uncertain

rad51d_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> rad51d_sa_blank
```

TP53
African
```{r}
tp53_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> tp53_afr_ben

tp53_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> tp53_afr_path

tp53_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> tp53_afr_uncertain

tp53_african_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="afr",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> tp53_afr_blank
```
European
```{r}
tp53_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> tp53_euro_ben

tp53_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> tp53_euro_path

tp53_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> tp53_euro_uncertain

tp53_european_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="euro",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> tp53_euro_blank
```
Latino
```{r}
tp53_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> tp53_lat_ben

tp53_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> tp53_lat_path

tp53_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> tp53_lat_uncertain

tp53_latino_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="lat",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> tp53_lat_blank
```
East Asia
```{r}
tp53_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign") -> tp53_ea_ben

tp53_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic") -> tp53_ea_path

tp53_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain") -> tp53_ea_uncertain

tp53_east_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="ea",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance)) -> tp53_ea_blank
```
South Asia
```{r}
tp53_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Benign")  -> tp53_sa_ben

tp53_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Pathogenic")  -> tp53_sa_path

tp53_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(clin_var_clinical_significance=="Uncertain")  -> tp53_sa_uncertain

tp53_south_asian_unique() %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(Population="sa",.after=chromosome) %>% 
   
  filter(is.na(clin_var_clinical_significance))  -> tp53_sa_blank
```

Merging all unique variants into 1 file
```{r}
full_join(atm_afr_ben, atm_afr_path) %>% full_join(atm_afr_uncertain) %>% full_join(atm_afr_blank) %>% full_join(atm_euro_ben) %>% full_join(atm_euro_path) %>% full_join(atm_euro_blank) %>% full_join(atm_euro_uncertain) %>% full_join(atm_lat_ben) %>% full_join(atm_lat_blank) %>% full_join(atm_lat_path) %>% full_join(atm_lat_uncertain) %>% full_join(atm_ea_ben) %>% full_join(atm_ea_blank) %>% full_join(atm_ea_uncertain) %>% full_join(atm_ea_path) %>% full_join(atm_sa_ben) %>% full_join(atm_sa_blank) %>% full_join(atm_sa_uncertain) %>% full_join(atm_sa_path) -> atm_unique

full_join(bard1_afr_ben, bard1_afr_path) %>% full_join(bard1_afr_uncertain) %>% full_join(bard1_afr_blank) %>% full_join(bard1_euro_ben) %>% full_join(bard1_euro_path) %>% full_join(bard1_euro_blank) %>% full_join(bard1_euro_uncertain) %>% full_join(bard1_lat_ben) %>% full_join(bard1_lat_blank) %>% full_join(bard1_lat_path) %>% full_join(bard1_lat_uncertain) %>% full_join(bard1_ea_ben) %>% full_join(bard1_ea_blank) %>% full_join(bard1_ea_uncertain) %>% full_join(bard1_ea_path) %>% full_join(bard1_sa_ben) %>% full_join(bard1_sa_blank) %>% full_join(bard1_sa_uncertain) %>% full_join(bard1_sa_path) -> bard1_unique

full_join(brca1_afr_ben, brca1_afr_path) %>% full_join(brca1_afr_uncertain) %>% full_join(brca1_afr_blank) %>% full_join(brca1_euro_ben) %>% full_join(brca1_euro_path) %>% full_join(brca1_euro_blank) %>% full_join(brca1_euro_uncertain) %>% full_join(brca1_lat_ben) %>% full_join(brca1_lat_blank) %>% full_join(brca1_lat_path) %>% full_join(brca1_lat_uncertain) %>% full_join(brca1_ea_ben) %>% full_join(brca1_ea_blank) %>% full_join(brca1_ea_uncertain) %>% full_join(brca1_ea_path) %>% full_join(brca1_sa_ben) %>% full_join(brca1_sa_blank) %>% full_join(brca1_sa_uncertain) %>% full_join(brca1_sa_path) -> brca1_unique

full_join(brca2_afr_ben, brca2_afr_path) %>% full_join(brca2_afr_uncertain) %>% full_join(brca2_afr_blank) %>% full_join(brca2_euro_ben) %>% full_join(brca2_euro_path) %>% full_join(brca2_euro_blank) %>% full_join(brca2_euro_uncertain) %>% full_join(brca2_lat_ben) %>% full_join(brca2_lat_blank) %>% full_join(brca2_lat_path) %>% full_join(brca2_lat_uncertain) %>% full_join(brca2_ea_ben) %>% full_join(brca2_ea_blank) %>% full_join(brca2_ea_uncertain) %>% full_join(brca2_ea_path) %>% full_join(brca2_sa_ben) %>% full_join(brca2_sa_blank) %>% full_join(brca2_sa_uncertain) %>% full_join(brca2_sa_path) -> brca2_unique

full_join(chek2_afr_ben, chek2_afr_path) %>% full_join(chek2_afr_uncertain) %>% full_join(chek2_afr_blank) %>% full_join(chek2_euro_ben) %>% full_join(chek2_euro_path) %>% full_join(chek2_euro_blank) %>% full_join(chek2_euro_uncertain) %>% full_join(chek2_lat_ben) %>% full_join(chek2_lat_blank) %>% full_join(chek2_lat_path) %>% full_join(chek2_lat_uncertain) %>% full_join(chek2_ea_ben) %>% full_join(chek2_ea_blank) %>% full_join(chek2_ea_uncertain) %>% full_join(chek2_ea_path) %>% full_join(chek2_sa_ben) %>% full_join(chek2_sa_blank) %>% full_join(chek2_sa_uncertain) %>% full_join(chek2_sa_path) -> chek2_unique

full_join(palb2_afr_ben, palb2_afr_path) %>% full_join(palb2_afr_uncertain) %>% full_join(palb2_afr_blank) %>% full_join(palb2_euro_ben) %>% full_join(palb2_euro_path) %>% full_join(palb2_euro_blank) %>% full_join(palb2_euro_uncertain) %>% full_join(palb2_lat_ben) %>% full_join(palb2_lat_blank) %>% full_join(palb2_lat_path) %>% full_join(palb2_lat_uncertain) %>% full_join(palb2_ea_ben) %>% full_join(palb2_ea_blank) %>% full_join(palb2_ea_uncertain) %>% full_join(palb2_ea_path) %>% full_join(palb2_sa_ben) %>% full_join(palb2_sa_blank) %>% full_join(palb2_sa_uncertain) %>% full_join(palb2_sa_path) -> palb2_unique

full_join(rad51c_afr_ben, rad51c_afr_path) %>% full_join(rad51c_afr_uncertain) %>% full_join(rad51c_afr_blank) %>% full_join(rad51c_euro_ben) %>% full_join(rad51c_euro_path) %>% full_join(rad51c_euro_blank) %>% full_join(rad51c_euro_uncertain) %>% full_join(rad51c_lat_ben) %>% full_join(rad51c_lat_blank) %>% full_join(rad51c_lat_path) %>% full_join(rad51c_lat_uncertain) %>% full_join(rad51c_ea_ben) %>% full_join(rad51c_ea_blank) %>% full_join(rad51c_ea_uncertain) %>% full_join(rad51c_ea_path) %>% full_join(rad51c_sa_ben) %>% full_join(rad51c_sa_blank) %>% full_join(rad51c_sa_uncertain) %>% full_join(rad51c_sa_path) -> rad51c_unique

full_join(rad51d_afr_ben, rad51d_afr_path) %>% full_join(rad51d_afr_uncertain) %>% full_join(rad51d_afr_blank) %>% full_join(rad51d_euro_ben) %>% full_join(rad51d_euro_path) %>% full_join(rad51d_euro_blank) %>% full_join(rad51d_euro_uncertain) %>% full_join(rad51d_lat_ben) %>% full_join(rad51d_lat_blank) %>% full_join(rad51d_lat_path) %>% full_join(rad51d_lat_uncertain) %>% full_join(rad51d_ea_ben) %>% full_join(rad51d_ea_blank) %>% full_join(rad51d_ea_uncertain) %>% full_join(rad51d_ea_path) %>% full_join(rad51d_sa_ben) %>% full_join(rad51d_sa_blank) %>% full_join(rad51d_sa_uncertain) %>% full_join(rad51d_sa_path) -> rad51d_unique

full_join(tp53_afr_ben, tp53_afr_path) %>% full_join(tp53_afr_uncertain) %>% full_join(tp53_afr_blank) %>% full_join(tp53_euro_ben) %>% full_join(tp53_euro_path) %>% full_join(tp53_euro_blank) %>% full_join(tp53_euro_uncertain) %>% full_join(tp53_lat_ben) %>% full_join(tp53_lat_blank) %>% full_join(tp53_lat_path) %>% full_join(tp53_lat_uncertain) %>% full_join(tp53_ea_ben) %>% full_join(tp53_ea_blank) %>% full_join(tp53_ea_uncertain) %>% full_join(tp53_ea_path) %>% full_join(tp53_sa_ben) %>% full_join(tp53_sa_blank) %>% full_join(tp53_sa_uncertain) %>% full_join(tp53_sa_path) -> tp53_unique

full_join(atm_unique, bard1_unique) %>% full_join(brca1_unique) %>% full_join(brca2_unique) %>% full_join(chek2_unique) %>% full_join(palb2_unique) %>% full_join(rad51c_unique) %>% full_join(rad51d_unique) %>% full_join(tp53_unique) -> gnomad_unique_all

gnomad_unique_all %>% write.csv(file = here("output/csv files/Unique variants.csv"), row.names = F)
```
Allele frequency table
```{r}
gnomad_freq <- gnomad_unique_all %>% 
  rename("gene"="chromosome")%>%
  mutate(chromosome = case_when(gene=="atm" ~ 11,
                                gene=="bard1" ~ 2,
                                gene=="brca1" ~ 17,
                                gene=="brca2" ~ 13,
                                gene=="chek2" ~ 22,
                                gene=="palb2" ~ 16,
                                gene=="rad51c" ~ 17,
                                gene=="rad51d" ~ 17,
                                gene=="tp53" ~ 17),.after = gene,
         Population = case_when(Population=="afr" ~ "African_american",
                                Population=="lat" ~ "Latino_admixed_american",
                                Population=="euro" ~ "Non_finnish_european",
                                Population=="ea" ~ "East_asian",
                                Population=="sa" ~ "South_asian"),
         clin_var_clinical_significance=case_when(is.na(clin_var_clinical_significance)~"Blank",
                                                  TRUE~clin_var_clinical_significance)) %>% 
  mutate(freq_allele = ifelse(allele_count_african_african_american!=0,
                              allele_count/allele_number_african_african_american,
                              ifelse(allele_count_latino_admixed_american!=0,
                                     allele_count/allele_number_latino_admixed_american,
                                     ifelse(allele_count_european_non_finnish!=0,
                                            allele_count/allele_number_european_non_finnish,
                                            ifelse(allele_count_east_asian!=0,
                                                   allele_count/allele_number_east_asian,
                                                   ifelse(allele_count_south_asian!=0,
                                                          allele_count/allele_number_south_asian,0)))))
         ,.after = allele_number) %>% 
  mutate(freq_log = (log10(freq_allele)),.after=freq_allele)
```
Gene distribution
```{r}
gnomad_freq %>% 
  group_by(gene) %>% 
  count(clin_var_clinical_significance) %>% 
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>% 
  replace_na(list(Pathogenic=0)) %>%
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>%
  rename("Gene" = "gene", "Unreported" = "Blank") %>% 
  write.csv(here("output/gene_distribution.csv"), row.names = F)

gnomad_freq %>% 
  group_by(gene) %>% 
  count(Population) %>% 
  pivot_wider(names_from = Population, values_from = n) %>% 
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>%
  rename("Gene" = "gene") %>% 
  write.csv(here("output/pop_distribution.csv"), row.names = F)

gnomad_freq %>% 
  group_by(gene) %>% 
  count(Population) %>%
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>%
  rename("Gene" = "gene") %>% 
  mutate(percent = round((n/sum(n))*100,2)) %>%
  ggplot(aes(x=Gene, y=percent, fill=Population)) + 
  geom_col(alpha = 0.7, colour="black") + 
  scale_fill_manual(values = c("African_american" = "#800080", 
                               "Latino_admixed_american" = "darkgreen", 
                               "Non_finnish_european" = "gray", 
                               "East_asian" = "navy",
                               "South_asian" = "darkred")) +
  labs(title = "Percentage of population unique variants for each gene", y="Percentage")
ggsave(here("output/pop.png"), height = 6, width = 8, dpi = 300)

gnomad_freq %>% 
  group_by(gene) %>% 
  mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", 
                                  vep_annotation %in% Inframe ~ "inframe_variant",
                                  vep_annotation%in%Stop ~ "stop_variant", 
                                  TRUE ~ vep_annotation)) %>% 
  count(vep_annotation) %>% 
  pivot_wider(names_from = vep_annotation, values_from = n) %>%
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>% 
  write.csv(here("output/vep_distribution.csv"), row.names = F)

gnomad_freq %>% 
  filter(clin_var_clinical_significance=="Blank") %>% 
  group_by(gene) %>% 
  mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", 
                                  vep_annotation %in% Inframe ~ "inframe_variant",
                                  vep_annotation%in%Stop ~ "stop_variant", 
                                  TRUE ~ vep_annotation)) %>% 
  count(vep_annotation) %>% 
  pivot_wider(names_from = vep_annotation, values_from = n) %>%
  replace_na(list(inframe_variant=0, stop_variant=0, start_lost=0)) %>% 
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>% 
  rename("Gene"="gene") %>%
  write.csv(here("output/novep_distribution.csv"), row.names = F)

gnomad_freq %>% 
  filter(clin_var_clinical_significance=="Blank") %>% 
  group_by(gene) %>% 
  mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", 
                                  vep_annotation %in% Inframe ~ "inframe_variant",
                                  vep_annotation%in%Stop ~ "stop_variant", 
                                  TRUE ~ vep_annotation)) %>% 
  count(vep_annotation) %>% 
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>% 
  rename("Gene"="gene") %>%
  ggplot(aes(x=Gene, y=n, fill=vep_annotation)) +
  geom_col(alpha = 0.7, colour="black") +
  labs(title = "Number of unreported frameshift population unique variants for each gene", y="Count")
ggsave(here("output/pop_unreported.png"), height = 6, width = 8, dpi = 300)

gnomad_freq %>% 
  filter(clin_var_clinical_significance=="Blank" , vep_annotation=="frameshift_variant") %>% 
  group_by(gene) %>% 
  count(Population) %>% 
  mutate(gene = case_when(gene =="atm" ~ "ATM",
                          gene =="bard1" ~ "BARD1",
                          gene =="brca1" ~ "BRCA1",
                          gene =="brca2" ~ "BRCA2",
                          gene =="chek2" ~ "CHEK2",
                          gene =="palb2" ~ "PALB2",
                          gene =="rad51c" ~ "RAD51C",
                          gene =="rad51d" ~ "RAD51D",
                          gene =="tp53" ~ "TP53")) %>% 
  rename("Gene"="gene") %>% 
  mutate(percent = round((n/sum(n))*100,2)) %>% 
  ggplot(aes(x=Gene, y=percent, fill=Population)) +
  geom_col(alpha = 0.7, colour="black") + 
  scale_fill_manual(values = c("African_american" = "#800080", 
                               "Latino_admixed_american" = "darkgreen", 
                               "Non_finnish_european" = "gray", 
                               "East_asian" = "navy",
                               "South_asian" = "darkred")) +
  labs(title = "Percentage of unreported frameshift population unique variants for each gene", y="Percentage")
ggsave(here("output/pop_unreported.png"), height = 6, width = 8, dpi = 300)
  
```
Frequency of non-annotated variants
```{r}
gnomad_freq %>% 
  unite(condition, gene, Population, sep = " ") %>% 
  filter(clin_var_clinical_significance=="Blank") %>% 
  group_by(condition) %>% count(clin_var_clinical_significance) %>% 
  separate(condition, into = c("gene", "Population"), sep=" ") %>% 
  group_by(gene) %>% 
  mutate(frequency= round((n/sum(n))*100,3)) %>% 
  select(-clin_var_clinical_significance) %>% 
  write.csv(file=here("output/csv files/unreported freqencies.csv"))
```
Frequency of pathogenic variants 
```{r}
#Between population
gnomad_freq %>% 
  unite(condition, gene, Population, sep = " ") %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  group_by(condition) %>% count(clin_var_clinical_significance) %>% 
  separate(condition, into = c("gene", "Population"), sep=" ") %>% 
  group_by(gene) %>% 
  mutate(frequency= round((n/sum(n))*100,3)) %>% 
  select(-clin_var_clinical_significance) %>% 
  write.csv(file=here("output/csv files/pathogenic_btw_freqencies.csv"))

#Within population
gnomad_freq %>% 
  unite(condition, gene, Population, sep = " ") %>% 
  group_by(condition) %>% count(clin_var_clinical_significance) %>% 
  mutate(frequency= round((n/sum(n))*100,3)) %>% 
  mutate(total = sum(n)) %>% 
  separate(condition , c("gene", "population"), sep =" ") %>% 
  filter(clin_var_clinical_significance == "Pathogenic") %>% 
  select(-clin_var_clinical_significance) %>% 
  write.csv(file=here("output/csv files/pathogenic_wth_freqencies.csv"))

#Missense
gnomad_freq %>% 
  unite(condition, gene, Population, sep = " ") %>% 
  group_by(condition) %>% 
  count(vep_annotation) %>% 
  mutate(frequency= round((n/sum(n))*100,3)) %>% 
  mutate(total = sum(n)) %>% 
  separate(condition , c("gene", "population"), sep =" ") %>% 
  filter(vep_annotation == "missense_variant") %>% 
  select(-vep_annotation)

gnomad_freq %>% 
  filter(vep_annotation=="missense_variant" & clin_var_clinical_significance!="Blank") %>% 
  unite(condition, gene, Population, sep = " ") %>% 
  group_by(condition) %>% 
  count(clin_var_clinical_significance) %>% 
  mutate(frequency= round((n/sum(n))*100,3)) %>% 
  mutate(total = sum(n)) %>% 
  separate(condition , c("gene", "population"), sep =" ") %>% 
  filter(clin_var_clinical_significance == "Pathogenic") %>% 
  select(-clin_var_clinical_significance)
```

