Library of packages
```{r}
library(tidyverse)
library(here)
library(janitor)
library(visdat)
library(skimr)
library(rstatix)
library(dplyr)
library(knitr)
library('corrr')
library(ggcorrplot)
library("FactoMineR")
library(factoextra)
library(rtracklayer)
library(GenomicRanges)
library(ggExtra)
```

Loading of gene data from gnomAD v4
```{r}
ATMv4 <- clean_names(read.csv(here("data/v4/ATM.csv"))) %>% mutate(gene = "ATM",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
BARD1v4 <- clean_names(read.csv(here("data/v4/BARD1.csv"))) %>% mutate(gene = "BARD1",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
BRCA1v4 <- clean_names(read.csv(here("data/v4/BRCA1.csv"))) %>% mutate(gene = "BRCA1",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
BRCA2v4 <- clean_names(read.csv(here("data/v4/BRCA2.csv"))) %>% mutate(gene = "BRCA2",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
CHEK2v4 <- clean_names(read.csv(here("data/v4/CHEK2.csv"))) %>% mutate(gene = "CHEK2",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
PALB2v4 <- clean_names(read.csv(here("data/v4/PALB2.csv"))) %>% mutate(gene = "PALB2",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
RAD51Cv4 <- clean_names(read.csv(here("data/v4/RAD51C.csv"))) %>% mutate(gene = "RAD51C",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
RAD51Dv4 <- clean_names(read.csv(here("data/v4/RAD51D.csv"))) %>% mutate(gene = "RAD51D",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))
TP53v4 <- clean_names(read.csv(here("data/v4/TP53.csv"))) %>% mutate(gene = "TP53",.before = chromosome) %>% replace_na(list("allele_count_amish" = 0))

Pathogenic <- c("Pathogenic", "Pathogenic/Likely pathogenic", "Likely pathogenic")
Benign <- c("Benign", "Benign/Likely benign", "Likely benign")
Uncertain <- c("Uncertain significance", "Conflicting interpretations of pathogenicity", "Conflicting interpretations of pathogenicity; risk factor")
Splice <- c("splice_donor_variant", "splice_acceptor_variant")
Inframe <- c("inframe_deletion", "inframe_insertion")
Stop <- c("stop_gained", "stop_lost")
```

Population and gene distribution of all variants and pathogenic variants
```{r}
gnomadv4_total <- gnomadv4_count %>% group_by(gene) %>% count(population) %>% rename("total" = "n")

gnomadv4_path <- gnomadv4_count %>% filter(clin_var_clinical_significance=="Pathogenic") %>% group_by(gene) %>% count(population) %>% rename("pathogenic_variants"= "n") %>% ungroup() %>% complete(gene, population, fill = list(n = 0)) %>% mutate(pathogenic_variants = case_when(is.na(pathogenic_variants) ~ 0, T ~ pathogenic_variants))
```

Figures using gnomAD v4 data
```{r}
gnomadv4_count %>% 
  group_by(gene) %>% 
  count(population) %>% 
  ggplot(aes(x= population, y = n, fill = gene)) + geom_col(colour = "#000000") + 
  theme_bw() +
  theme(axis.title.y = element_blank(),
        legend.title = element_blank(),
        axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        legend.text = element_text(size = 15)) +
  scale_fill_manual(values = c("ATM" = "darkgreen",
                               "BARD1" = "violet",
                               "BRCA1" = "darkred",
                               "BRCA2" = "maroon",
                               "CHEK2" = "skyblue",
                               "PALB2" = "lightyellow",
                               "RAD51C" = "darkorange",
                               "RAD51D" = "orange",
                               "TP53" = "#ffcccc")) +
  scale_y_continuous(breaks = seq(0, 10000 ,by = 500)) + 
  xlim("NFE", "AFR","AMR", "EAS", "SAS") +
  labs(x="Population")
ggsave(here("output/v4/fig1B.png"), dpi=300, width=8, height=6)

gnomadv4_count %>% 
  group_by(gene) %>% 
  count(clin_var_clinical_significance) %>% 
  mutate(percent = round((n/sum(n))*100,2)) %>% 
  ggplot(aes(x=gene, y=percent, fill=clin_var_clinical_significance)) + 
  geom_col(colour="black") + 
  scale_fill_manual(values = c("Benign" = "skyblue", 
                                 "Pathogenic" = "red", 
                                 "Unreported" = "darkgray", 
                                 "Uncertain" = "darkgreen")) +
  theme_bw() + 
  theme(axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15, angle = 30, hjust=1),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.text = element_text(size = 15),
        legend.title = element_blank()) +
  labs(x="Gene", y = "Percentage of variants")
ggsave(here("output/v4/fig1C.png"), height = 6, width = 8, dpi = 300)

gnomadv4_count %>%
  filter(clin_var_clinical_significance=="Unreported") %>% 
  group_by(gene) %>%
  count(vep_annotation) %>%
  mutate(percent = round((n/sum(n))*100,2),
         vep_annotation = case_when(vep_annotation == "frameshift_variant" ~ "Frameshift Variant",
                                    vep_annotation == "inframe_variant" ~ "Inframe Variant",
                                    vep_annotation == "missense_variant" ~ "Missense Variant",
                                    vep_annotation == "splice_site_variant" ~ "Splice Site Variant",
                                    vep_annotation == "start_lost" ~ "Start Lost",
                                    vep_annotation == "stop_gained" ~ "Stop Gained",
                                    vep_annotation == "stop_lost" ~ "Stop Lost")) %>% 
  ggplot(aes(x=gene, y=percent, fill=vep_annotation)) + 
  geom_col(colour="black") +
  theme_bw() + 
  theme(axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15, angle = 30, hjust=1),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.text = element_text(size = 15),
        legend.title = element_text(size = 18)) +
  scale_fill_manual(values = c("Frameshift Variant" = "darkred",
                               "Inframe Variant" = "darkorange",
                               "Missense Variant" = "darkgrey",
                               "Splice Site Variant" = "skyblue",
                               "Start Lost" = "cyan",
                               "Stop Gained" = "blue",
                               "Stop Lost" = "gold")) +
  labs(x = "Gene", y = "Percentage of variants", fill = "Types of Variants")
ggsave(here("output/v4/fig1D.png"), height = 6, width = 8, dpi = 300)

gnomadv4_count %>% 
  unite(test, gene, clin_var_clinical_significance, sep = " ") %>% 
  group_by(test) %>%
  count(count_range) %>% 
  mutate(n= case_when(is.na(n)~ 0, TRUE~n)) %>% 
  mutate(normalised = round((n / sum(n)) * 100, 2)) %>% 
  separate(test, into = c("gene", "clin_sig"), sep = " ") %>% 
    ggplot(aes(fill=factor(count_range, c("0-1", "2-3", "4-10", "11-100", "100+")) , x = clin_sig, y = normalised)) + 
    facet_wrap(~gene) +
    geom_col(alpha = 0.7, colour="black") + 
    scale_fill_manual(values = c("0-1" = "skyblue", 
                                 "2-3" = "blue", 
                                 "4-10" = "darkblue", 
                                 "11-100" = "purple",
                                 "100+" = "red")) +
  theme_bw() + 
    theme(axis.text.y = element_text(size = 15),
          axis.text.x = element_text(size = 15,angle = 30, hjust=1),
          axis.title.x = element_text(size = 20),
          axis.title.y = element_text(size = 20),
          legend.text = element_text(size = 15),
          legend.title = element_text(size = 20),
          strip.text = element_text(size = 10.5)) +
    xlim("Benign", "Pathogenic", "Uncertain", "Unreported") +
    labs(fill = "Allele count range", x = "ClinVar Clinical Significance", y = "Percentage of variants")
ggsave(here("output/v4/fig1E.png"), height = 6, width = 8, dpi = 300)

gnomadv4_count %>% 
  group_by(population) %>% 
  count(clin_var_clinical_significance) %>% 
  mutate(percent = round((n/sum(n))*100,2)) %>% 
  ggplot(aes(x=population, y=percent, fill=clin_var_clinical_significance)) + 
  geom_col(colour="black") + 
  scale_fill_manual(values = c("Benign" = "skyblue", 
                                 "Pathogenic" = "red", 
                                 "Unreported" = "darkgray", 
                                 "Uncertain" = "darkgreen")) +
  theme_bw() + 
  theme(axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.text = element_text(size = 15),
        legend.title = element_blank()) +
  xlim("NFE", "AFR", "AMR", "EAS", "SAS") +
  labs(x="Population", y = "Percentage of variants")
ggsave(here("output/v4/fig1F.png"), height = 6, width = 8, dpi = 300)

gnomad_count %>%
  mutate(Population = case_when(Population == "African_american" ~ "AFR",
                                Population == "Latino_admixed_american" ~ "AMR",
                                Population == "Non_finnish_european" ~ "NFE", 
                                Population == "East_asian" ~ "EAS",
                                Population == "South_asian" ~ "SAS"),
         gene = case_when(gene == "atm" ~ "ATM",
                          gene == "bard1" ~ "BARD1",
                          gene == "brca1" ~ "BRCA1",
                          gene == "brca2" ~ "BRCA2",
                          gene == "chek2" ~ "CHEK2",
                          gene == "palb2" ~ "PALB2",
                          gene == "rad51c" ~ "RAD51C",
                          gene == "rad51d" ~ "RAD51D",
                          gene == "tp53" ~ "TP53")) %>% 
  unite(test, gene, Population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("Blank", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "Population"), sep = " ") %>% 
  full_join(gnomad_total) %>% 
  mutate(v2 = round((in_clinvar/total)*100,2)) %>% 
  select(gene, Population, v2) %>% 
  full_join(gnomadv4_inclinvar) %>% 
  pivot_longer(cols = c("v2", "v4"), names_to = "Version", values_to = "Percentage") %>% 
  ggplot(aes(x=Population, y=Percentage, shape=Population)) +
  facet_wrap(~gene) +
  geom_point(size = 2.5) + geom_line(aes(group = Version, color=Version), linetype = 2, size = 0.7) +
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        axis.text.y = element_text(size = 15),
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 15)) + 
  ylim(0, 100) + xlim("NFE", "AFR", "AMR", "EAS", "SAS")
ggsave(here("output/v4/fig2A.png"), width = 8, height = 6, dpi = 300)

gnomadv4_combine_total <- gnomadv4_count %>% filter(vep_annotation=="frameshift_variant"|vep_annotation=="splice_site_variant"|vep_annotation=="stop_lost") %>% group_by(gene) %>% count(population) %>% rename("total" = "n")

gnomadv4_count %>%
  filter(vep_annotation=="frameshift_variant"|vep_annotation=="splice_site_variant"|vep_annotation=="stop_lost") %>% 
  unite(test, gene, population, sep = " ") %>% 
  group_by(test) %>% 
  count(clin_var_clinical_significance) %>% 
  ungroup() %>% 
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("unreported", clin_var_clinical_significance, ignore.case = TRUE)) %>% 
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "population"), sep = " ") %>% 
  full_join(gnomadv4_combine_total) %>% 
  mutate(percent_in_clinvar = round((in_clinvar/total)*100,2)) %>% 
  ggplot(aes(x=population,y=percent_in_clinvar, shape=population)) + geom_point(size = 2.5) + 
  geom_line(aes(group = gene), color = "black", linetype = 2, size = 0.6) +
  facet_wrap(~gene) +
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        axis.text.y = element_text(size = 15),
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) + 
  ylim(0, 100) + xlim("NFE", "AFR", "AMR", "EAS", "SAS") +
  labs(x = "Population", y = "Percentage", shape = "Population")
ggsave(here("output/v4/fig2B.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(cadd_raw_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = cadd_raw_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 20),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "CADD raw Rankscore")
ggsave(here("output/v4/fig3A.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(alpha_missense_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = alpha_missense_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 20),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "Alpha Missense Rankscore")
ggsave(here("output/v4/fig3B.png"), width = 8, height = 6, dpi = 300)

prediction_v4 %>%
  unite(test, gene, clin_var_clinical_significance, sep = " ") %>%
  group_by(test) %>%
  count(count) %>%
  complete(count = full_seq(0:7, 1), fill = list(n = 0)) %>%
  mutate(n= case_when(is.na(n)~ 0, TRUE~n)) %>% 
  mutate(normalised = round((n / sum(n))*100, 2)) %>% 
  separate(test, into = c("gene", "Clin_var_significance"), sep = " ") %>%
  ggplot(aes(x=as.character(count), y=normalised, fill=Clin_var_significance)) + facet_wrap(~gene) + geom_col(position = "dodge") + ylim(0,100) + labs(x = "Number of pathogenic predictions", y= "Percentage")+
  scale_fill_manual(values = c("Benign" = "skyblue", "Pathogenic" = "red", "Unreported" = "darkgray", "Uncertain" = "darkgreen")) +
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 18),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) 
ggsave(here("output/v4/fig3C.png"), width = 8, height = 6, dpi = 300)

pca_v4_gnomad <- ggplot(pca_gnomadv4_modified_rank, aes(x = V2, y = V3, colour = clin_sig)) +
  scale_colour_manual(values = c("Benign" = "skyblue", "Pathogenic" = "red", "Unreported" = "darkgray", "Uncertain" = "darkgreen")) + theme_bw() +
  labs(x = "Principal Component 1", y = "Principal Component 2") +
  theme(axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.text = element_text(size = 15),
        legend.title = element_blank()) +
  geom_point() +
  geom_point(data = filter(pca_gnomadv4_modified, clin_sig=="Unreported")) +
  geom_point(data = filter(pca_gnomadv4_modified, clin_sig=="Uncertain")) +
  geom_point(data = filter(pca_gnomadv4_modified, clin_sig=="Benign"), size = 2) +
  geom_point(data = filter(pca_gnomadv4_modified, clin_sig=="Pathogenic"), size = 2)

ggMarginal(pca_v4_gnomad, type = "boxplot", margins = "both", groupFill = T, groupColour = T, xparams = )
ggsave(here("output/v4/fig3D.png"), width = 8, height = 6, dpi = 300)
```

Supplementary figures of rankscores
```{r}
dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(cadd_raw_rankscore_hg19)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = cadd_raw_rankscore_hg19)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "CADD raw hg19 Rankscore")
ggsave(here("output/v4/ranks/all/cadd_raw_hg19.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(revel_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = revel_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "Revel Rankscore")
ggsave(here("output/v4/ranks/all/revel.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(meta_lr_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = meta_lr_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "MetaLR Rankscore")
ggsave(here("output/v4/ranks/all/meta_lr.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(meta_svm_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = meta_svm_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "MetaSVM Rankscore")
ggsave(here("output/v4/ranks/all/meta_svm.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(vest4_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = vest4_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "VEST4 Rankscore")
ggsave(here("output/v4/ranks/all/vest4.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(gerp_rs_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = gerp_rs_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "GERP++ RS rankscore")
ggsave(here("output/v4/ranks/all/gerp.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(sift_converted_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = sift_converted_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "SIFT Rankscore")
ggsave(here("output/v4/ranks/all/sift.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(polyphen2_hdiv_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = polyphen2_hdiv_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "PolyPhen2 HDIV Rankscore")
ggsave(here("output/v4/ranks/all/polyphen2_hdiv.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(polyphen2_hvar_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = polyphen2_hvar_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "PolyPhen2 HVAR Rankscore")
ggsave(here("output/v4/ranks/all/PolyPhen2_HVAR.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(mutation_assessor_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = mutation_assessor_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "Mutation Assessor Rankscore")
ggsave(here("output/v4/ranks/all/mutation_assessor.png"), width = 8, height = 6, dpi = 300)

dbnsfp_v4gnomad %>% 
  rename("gene"="gene.x") %>% 
  filter(!is.na(fathmm_converted_rankscore)) %>% 
  ggplot(aes(x = clin_var_clinical_significance, y = fathmm_converted_rankscore)) + 
  geom_violin(fill = "#0AD6F2", alpha = 0.4, width = 0.9) +
  facet_wrap("gene") +
  stat_summary(fun = "median", geom = "point", colour = "#0A2581", position = position_dodge(0.3)) +
  stat_summary(fun.data = "median_hilow", geom = "errorbar", colour = "#0A2581", width=0.3) +
  ylim(0,1) +
  xlim("Benign", "Pathogenic", "Uncertain", "Unreported") + 
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 10.5)) +
  labs(x="ClinVar Clinical Significance", y = "FATHMM Rankscore")
ggsave(here("output/v4/ranks/all/fathmm_rankscore.png"), width = 8, height = 6, dpi = 300)
```

Comparison of v2 and v4
```{r}
gnomad_total <- gnomad_count %>% 
  mutate(Population = case_when(Population == "African_american" ~ "AFR",
                                Population == "Latino_admixed_american" ~ "AMR",
                                Population == "Non_finnish_european" ~ "NFE", 
                                Population == "East_asian" ~ "EAS",
                                Population == "South_asian" ~ "SAS"),
         gene = case_when(gene == "atm" ~ "ATM",
                          gene == "bard1" ~ "BARD1",
                          gene == "brca1" ~ "BRCA1",
                          gene == "brca2" ~ "BRCA2",
                          gene == "chek2" ~ "CHEK2",
                          gene == "palb2" ~ "PALB2",
                          gene == "rad51c" ~ "RAD51C",
                          gene == "rad51d" ~ "RAD51D",
                          gene == "tp53" ~ "TP53")) %>% 
  group_by(gene) %>% count(Population) %>% rename("total" = "n")

gnomadv4_inclinvar <- gnomadv4_count %>%
  unite(test, gene, population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("unreported", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "population"), sep = " ") %>% 
  full_join(gnomadv4_total) %>% 
  mutate(v4 = round((in_clinvar/total)*100,2)) %>% 
  rename("Population" = "population") %>% 
  select(gene, Population, v4)

gnomad_count %>%
  mutate(Population = case_when(Population == "African_american" ~ "AFR",
                                Population == "Latino_admixed_american" ~ "AMR",
                                Population == "Non_finnish_european" ~ "NFE", 
                                Population == "East_asian" ~ "EAS",
                                Population == "South_asian" ~ "SAS"),
         gene = case_when(gene == "atm" ~ "ATM",
                          gene == "bard1" ~ "BARD1",
                          gene == "brca1" ~ "BRCA1",
                          gene == "brca2" ~ "BRCA2",
                          gene == "chek2" ~ "CHEK2",
                          gene == "palb2" ~ "PALB2",
                          gene == "rad51c" ~ "RAD51C",
                          gene == "rad51d" ~ "RAD51D",
                          gene == "tp53" ~ "TP53")) %>% 
  unite(test, gene, Population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("Blank", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "Population"), sep = " ") %>% 
  full_join(gnomad_total) %>% 
  mutate(v2 = round((in_clinvar/total)*100,2)) %>% 
  select(gene, Population, v2) %>% 
  full_join(gnomadv4_inclinvar) %>% 
  pivot_longer(cols = c("v2", "v4"), names_to = "Version", values_to = "Percentage") %>% 
  ggplot(aes(x=Population, y=Percentage, shape=Population)) +
  facet_wrap(~gene) +
  geom_point(size = 2.5) + geom_line(aes(group = Version, color=Version), linetype = 2, size = 0.7) +
  theme_bw() + 
  theme(axis.text.x = element_text(size = 15, angle = 35, hjust = 1),
        axis.title.x = element_text(size = 15),
        axis.title.y = element_blank(),
        axis.text.y = element_text(size = 15),
        legend.title = element_text(size = 15),
        legend.text = element_text(size = 15)) + 
  ylim(0, 100) + xlim("NFE", "AFR", "AMR", "EAS", "SAS")
ggsave(here("output/v4/fig2A.png"), width = 8, height = 6, dpi = 300)


comparison_count <- gnomadv4_count %>% 
                      group_by(gene) %>% 
                      count(population) %>% 
                      rename(v4=n)

gnomad_count %>% 
  mutate(Pop = case_when(Population == "African_american" ~ "AFR", 
                         Population == "Latino_admixed_american" ~ "AMR",
                         Population == "Non_finnish_european" ~ "NFE", 
                         Population == "East_asian" ~ "EAS", 
                         Population == "South_asian" ~ "SAS"),.after = Population) %>% 
  mutate(gene = case_when(gene == "atm" ~ "ATM",
                          gene == "bard1" ~ "BARD1",
                          gene == "brca1" ~ "BRCA1",
                          gene == "brca2" ~ "BRCA2",
                          gene == "chek2" ~ "CHEK2",
                          gene == "palb2" ~ "PALB2",
                          gene == "rad51c" ~ "RAD51C",
                          gene == "rad51d" ~ "RAD51D",
                          gene == "tp53" ~ "TP53")) %>% 
  group_by(gene) %>% 
  count(Pop) %>%
  rename(population=Pop, v2 = n) %>% 
  full_join(comparison_count) %>% 
  pivot_longer(cols = c("v2", "v4"), names_to = "version", values_to = "count") %>% 
  ggplot(aes(x= population, y = count, fill = gene)) + geom_col(position = "dodge") + facet_wrap(~version) +
  theme(axis.title.y = element_blank(),
        legend.title = element_blank(),
        axis.text.y = element_text(size = 15),
        axis.text.x = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        legend.text = element_text(size = 15)) +
  scale_fill_manual(values = c("ATM" = "darkgreen",
                               "BARD1" = "violet",
                               "BRCA1" = "darkred",
                               "BRCA2" = "maroon",
                               "CHEK2" = "skyblue",
                               "PALB2" = "lightyellow",
                               "RAD51C" = "darkorange",
                               "RAD51D" = "orange",
                               "TP53" = "#ffcccc")) +
  scale_y_continuous(breaks = seq(0, 3000, by = 500)) + 
  xlim("NFE", "AFR","AMR", "EAS", "SAS")
```

LiftOver -> hg19 to hg38 -> gnomad v2
```{r}
variants_v2 <- gnomad_count %>% add_column(chr="chr") %>% unite(chr, chr, chromosome, sep="") %>% select(chr, position)

chain_file_38 <- "D:/rcode/Rcode/work/Work/hg19ToHg38.over.chain" 
chain_38 <- import.chain(chain_file_38)

variants_gr <- GRanges(
  seqnames = variants_v2$chr,
  ranges = IRanges(start = variants_v2$position, end = variants_v2$position))

hg38_v2 <- liftOver(variants_gr, chain_38) %>% as.data.frame() %>% select(group, seqnames, end) %>% rename("id" = "group") %>% unite(id, id, seqnames)

hg38_pos <- variants_v2 %>% mutate(id = row_number()) %>% unite(id, id, chr) %>% inner_join(hg38_v2, by= "id") %>% rename("v2_hg38_pos" = "end", "v2_hg19_pos" = "position") %>% unite(id, id, v2_hg19_pos)

gnomadv2_hg38 <- gnomad_count %>% mutate(id = row_number(),.before = "gene") %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation), clin_var_clinical_significance = case_when(clin_var_clinical_significance=="Blank" ~ "Unreported", T ~ clin_var_clinical_significance)) %>% mutate(chr="chr") %>% unite(chr, chr, chromosome, sep = "") %>% unite(id, id, chr, position) %>% inner_join(hg38_pos, by = "id") %>% separate(id, into = c("id", "chromosome", "v2_hg19_pos")) %>% mutate(chromosome = case_when(chromosome=="chr11" ~ 11, chromosome=="chr13" ~ 13, chromosome=="chr16" ~ 16, chromosome=="chr17" ~ 17, chromosome=="chr2" ~ 2, chromosome=="chr22" ~ 22)) %>% unite(gnom_ad_id, chromosome, v2_hg38_pos, reference, alternate, sep = "-") %>% mutate(population_v2 = case_when(Population=="African_american" ~ "AFR", Population=="Latino_admixed_american" ~ "AMR", Population=="Non_finnish_european" ~ "NFE", Population=="East_asian" ~ "EAS", Population=="South_asian" ~ "SAS")) %>% select(gnom_ad_id, v2_hg19_pos, clin_var_clinical_significance, population_v2) %>% rename("clinvar_v2" = "clin_var_clinical_significance")
```
5769 variants

Variants from Gnomad v2 to v4
 - Population changes
```{r}
gnomadv4 %>% inner_join(gnomadv2_hg38, by = "gnom_ad_id") %>% mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(clin_var_clinical_significance = case_when(clin_var_clinical_significance== "" ~ "Unreported", is.na(clin_var_clinical_significance) ~ "Unreported", T ~ clin_var_clinical_significance)) %>% replace_na(list("allele_count_amish" = 0)) %>%  mutate(population_v4 = case_when(
      allele_count_african_african_american != 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AFR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american != 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AMR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian != 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "EAS",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish != 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "NFE",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian != 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "SAS",
      TRUE ~ "Shared")) %>% 
  group_by(population_v2) %>% 
  count(population_v4) %>% 
  mutate(percentage = round(n / sum(n) * 100,2))

gnomadv4 %>% inner_join(gnomadv2_hg38, by = "gnom_ad_id") %>% mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(clin_var_clinical_significance = case_when(clin_var_clinical_significance== "" ~ "Unreported", is.na(clin_var_clinical_significance) ~ "Unreported", T ~ clin_var_clinical_significance)) %>% replace_na(list("allele_count_amish" = 0)) %>% 
  mutate(population_v4 = case_when(
      allele_count_african_african_american != 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AFR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american != 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AMR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian != 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "EAS",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish != 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "NFE",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian != 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "SAS",
      TRUE ~ "Shared")) %>% 
  group_by(population_v2) %>% 
  count(population_v4) %>% 
  mutate(percentage = round(n / sum(n) * 100,2)) %>% 
  ggplot(aes(x = "", y = percentage, fill = population_v4)) +
  geom_bar(stat = "identity", width = 1) +
  coord_polar(theta = "y") +
  facet_wrap(~ population_v2, ncol = 5) +
  theme_void() +
  labs(fill = "Population v4", title = "Pie charts") +
  theme(legend.position = "bottom") + 
  scale_fill_manual(values = c("AFR" = "#800080", 
                               "AMR" = "darkgreen", 
                               "NFE" = "gray", 
                               "EAS" = "navy",
                               "SAS" = "darkred",
                               "Shared" = "darkorange"))
ggsave(here("output/v4/changes.png"), height = 10, width = 20, dpi = 300)

gnomadv4 %>% 
  inner_join(gnomadv2_hg38, by = "gnom_ad_id") %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% 
  mutate(clin_var_clinical_significance = case_when(clin_var_clinical_significance== "" ~ "Unreported", is.na(clin_var_clinical_significance) ~ "Unreported", T ~ clin_var_clinical_significance)) %>%
  replace_na(list("allele_count_amish" = 0)) %>% 
  mutate(population_v4 = case_when(
      allele_count_african_african_american != 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AFR",
      
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american != 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AMR",
      
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian != 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "EAS",
      
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish != 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "NFE",
      
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian != 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "SAS",
      TRUE ~ "Shared")) %>% 
  filter(population_v4 == "Shared") %>% 
  mutate(AFR = case_when(allele_count_african_african_american != 0 ~ "True"),
         AMR = case_when(allele_count_admixed_american != 0 ~ "True"),
         NFE = case_when(allele_count_european_non_finnish != 0 ~ "True"),
         EAS = case_when(allele_count_east_asian != 0 ~ "True"),
         SAS = case_when(allele_count_south_asian != 0 ~ "True"),
         ASJ = case_when(allele_count_ashkenazi_jewish != 0 ~ "True"),
         FIN = case_when(allele_count_european_finnish != 0 ~ "True"),
         MID = case_when(allele_count_middle_eastern != 0 ~ "True"),
         AMS = case_when(allele_count_amish != 0 ~ "True"),
         REM = case_when(allele_count_remaining != 0 ~ "True")) %>% 
  select(gnom_ad_id, v2_hg19_pos, population_v4, population_v2, clin_var_clinical_significance, clinvar_v2, AFR, AMR, NFE, EAS, SAS, ASJ, FIN, MID, AMS, REM)
```
 - ClinVar changes
```{r}
gnomadv4 %>% inner_join(gnomadv2_hg38, by = "gnom_ad_id") %>% mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(clin_var_clinical_significance = case_when(clin_var_clinical_significance== "" ~ "Unreported", is.na(clin_var_clinical_significance) ~ "Unreported", T ~ clin_var_clinical_significance)) %>% mutate(population_v4 = case_when(
      allele_count_african_african_american != 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AFR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american != 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "AMR",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian != 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "EAS",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish != 0 &
      allele_count_south_asian == 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "NFE",
      allele_count_african_african_american == 0 &
      allele_count_remaining == 0 &
      allele_count_admixed_american == 0 &
      allele_count_ashkenazi_jewish == 0 &
      allele_count_east_asian == 0 &
      allele_count_european_finnish == 0 &
      allele_count_european_non_finnish == 0 &
      allele_count_south_asian != 0 &
      allele_count_middle_eastern == 0 &
      allele_count_amish == 0 ~ "SAS",
      TRUE ~ "Shared")) %>% 
  select(gene, gnom_ad_id, v2_hg19_pos, clin_var_clinical_significance, clinvar_v2, population_v4, population_v2) %>% 
  group_by(clinvar_v2) %>% 
  count(clin_var_clinical_significance)
```

Fisher test
```{r}
gnomadv4_fisher <- gnomadv4_count %>%
  unite(test, gene, population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("unreported", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "population"), sep = " ") %>% 
  full_join(gnomadv4_path) %>% full_join(gnomadv4_total) %>%
  mutate(percent_pathogenic = round((pathogenic_variants/in_clinvar)*100,2),
         unreported = (total - in_clinvar))

#Function for calculating fisher exact test comparing pathogenicity between populations for genes in clinvar 
gnomadv4fisher_path <- function(x,y) {gnomadv4_fisher %>% 
  filter(gene == x, population %in% c(y, "NFE")) %>% 
  mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()}

#ATM
gnomadv4fisher_path("ATM", "AFR")
gnomadv4fisher_path("ATM", "AMR")
gnomadv4fisher_path("ATM", "EAS")
gnomadv4fisher_path("ATM", "SAS")

#BARD1
gnomadv4fisher_path("BARD1", "AFR")
gnomadv4fisher_path("BARD1", "AMR")
gnomadv4fisher_path("BARD1", "EAS")
gnomadv4fisher_path("BARD1", "SAS")

#BRCA1
gnomadv4fisher_path("BRCA1", "AFR")
gnomadv4fisher_path("BRCA1", "AMR")
gnomadv4fisher_path("BRCA1", "EAS")
gnomadv4fisher_path("BRCA1", "SAS")

#BRCA2
gnomadv4fisher_path("BRCA2", "AFR")
gnomadv4fisher_path("BRCA2", "AMR")
gnomadv4fisher_path("BRCA2", "EAS")
gnomadv4fisher_path("BRCA2", "SAS")

#CHEK2
gnomadv4fisher_path("CHEK2", "AFR")
gnomadv4fisher_path("CHEK2", "AMR")
gnomadv4fisher_path("CHEK2", "EAS")
gnomadv4fisher_path("CHEK2", "SAS")

#PALB2
gnomadv4fisher_path("PALB2", "AFR")
gnomadv4fisher_path("PALB2", "AMR")
gnomadv4fisher_path("PALB2", "EAS")
gnomadv4fisher_path("PALB2", "SAS")

#RAD51C
gnomadv4fisher_path("RAD51C", "AFR")
gnomadv4fisher_path("RAD51C", "AMR")
gnomadv4fisher_path("RAD51C", "EAS")
gnomadv4fisher_path("RAD51C", "SAS")

#RAD51D
gnomadv4fisher_path("RAD51D", "AFR")
gnomadv4fisher_path("RAD51D", "AMR")
gnomadv4fisher_path("RAD51D", "EAS")
gnomadv4fisher_path("RAD51D", "SAS")

#TP53
gnomadv4fisher_path("TP53", "AFR")
gnomadv4fisher_path("TP53", "AMR")
gnomadv4fisher_path("TP53", "EAS")
gnomadv4fisher_path("TP53", "SAS")

#All genes
gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

#Function for calculating fisher exact test comparing unreported between populations for genes in clinvar 
gnomadv4fisher_unr <- function(x,y) {gnomadv4_fisher %>% 
  filter(gene == x, population %in% c(y, "NFE")) %>% 
  mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()}

#ATM
gnomadv4fisher_unr("ATM", "AFR")
gnomadv4fisher_unr("ATM", "AMR")
gnomadv4fisher_unr("ATM", "EAS")
gnomadv4fisher_unr("ATM", "SAS")

#BARD1
gnomadv4fisher_unr("BARD1", "AFR")
gnomadv4fisher_unr("BARD1", "AMR")
gnomadv4fisher_unr("BARD1", "EAS")
gnomadv4fisher_unr("BARD1", "SAS")

#BRCA1
gnomadv4fisher_unr("BRCA1", "AFR")
gnomadv4fisher_unr("BRCA1", "AMR")
gnomadv4fisher_unr("BRCA1", "EAS")
gnomadv4fisher_unr("BRCA1", "SAS")

#BRCA2
gnomadv4fisher_unr("BRCA2", "AFR")
gnomadv4fisher_unr("BRCA2", "AMR")
gnomadv4fisher_unr("BRCA2", "EAS")
gnomadv4fisher_unr("BRCA2", "SAS")

#CHEK2
gnomadv4fisher_unr("CHEK2", "AFR")
gnomadv4fisher_unr("CHEK2", "AMR")
gnomadv4fisher_unr("CHEK2", "EAS")
gnomadv4fisher_unr("CHEK2", "SAS")

#PALB2
gnomadv4fisher_unr("PALB2", "AFR")
gnomadv4fisher_unr("PALB2", "AMR")
gnomadv4fisher_unr("PALB2", "EAS")
gnomadv4fisher_unr("PALB2", "SAS")

#RAD51C
gnomadv4fisher_unr("RAD51C", "AFR")
gnomadv4fisher_unr("RAD51C", "AMR")
gnomadv4fisher_unr("RAD51C", "EAS")
gnomadv4fisher_unr("RAD51C", "SAS")

#RAD51D
gnomadv4fisher_unr("RAD51D", "AFR")
gnomadv4fisher_unr("RAD51D", "AMR")
gnomadv4fisher_unr("RAD51D", "EAS")
gnomadv4fisher_unr("RAD51D", "SAS")

#TP53
gnomadv4fisher_unr("TP53", "AFR")
gnomadv4fisher_unr("TP53", "AMR")
gnomadv4fisher_unr("TP53", "EAS")
gnomadv4fisher_unr("TP53", "SAS")

#All genes
gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>%
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>%
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>%
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>%
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()
```

Fisher test for combination of Inframe, Frameshift, Splice Site and Stop Lost variants
```{r}
gnomadv4_total_comb <- gnomadv4_count %>%
  filter(vep_annotation=="inframe_variant"|vep_annotation=="splice_site_variant"|vep_annotation=="stop_gained"|
           vep_annotation=="frameshift_variant") %>% group_by(gene) %>% count(population) %>% rename("total" = "n")

gnomadv4_path_comb <- gnomadv4_count %>%
  filter(vep_annotation=="inframe_variant"|vep_annotation=="splice_site_variant"|vep_annotation=="stop_gained"|
           vep_annotation=="frameshift_variant") %>% filter(clin_var_clinical_significance=="Pathogenic") %>% group_by(gene) %>% count(population) %>% rename("pathogenic_variants"= "n") %>% ungroup() %>% complete(gene, population, fill = list(n = 0)) %>% mutate(pathogenic_variants = case_when(is.na(pathogenic_variants) ~ 0, T ~ pathogenic_variants))

gnomadv4_fisher_combination <- gnomadv4_count %>%
  filter(vep_annotation=="inframe_variant"|vep_annotation=="splice_site_variant"|vep_annotation=="stop_gained"|
           vep_annotation=="frameshift_variant") %>% 
  unite(test, gene, population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("unreported", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "population"), sep = " ") %>% 
  full_join(gnomadv4_path_comb) %>% full_join(gnomadv4_total_comb) %>%
  mutate(percent_pathogenic = round((pathogenic_variants/in_clinvar)*100,2),
         unreported = (total - in_clinvar))

#Function for calculating fisher exact test comparing pathogenicity between populations for genes in clinvar 
gnomadv4fisher_path_comb <- function(x,y) {gnomadv4_fisher_combination %>% 
  filter(gene == x, population %in% c(y, "NFE")) %>% 
  mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()}

#ATM
gnomadv4fisher_path_comb("ATM", "AFR")
gnomadv4fisher_path_comb("ATM", "AMR")
gnomadv4fisher_path_comb("ATM", "EAS")
gnomadv4fisher_path_comb("ATM", "SAS")

#BARD1
gnomadv4fisher_path_comb("BARD1", "AFR")
gnomadv4fisher_path_comb("BARD1", "AMR")
gnomadv4fisher_path_comb("BARD1", "EAS")
gnomadv4fisher_path_comb("BARD1", "SAS")

#BRCA1
gnomadv4fisher_path_comb("BRCA1", "AFR")
gnomadv4fisher_path_comb("BRCA1", "AMR")
gnomadv4fisher_path_comb("BRCA1", "EAS")
gnomadv4fisher_path_comb("BRCA1", "SAS")

#BRCA2
gnomadv4fisher_path_comb("BRCA2", "AFR")
gnomadv4fisher_path_comb("BRCA2", "AMR")
gnomadv4fisher_path_comb("BRCA2", "EAS")
gnomadv4fisher_path_comb("BRCA2", "SAS")

#CHEK2
gnomadv4fisher_path_comb("CHEK2", "AFR")
gnomadv4fisher_path_comb("CHEK2", "AMR")
gnomadv4fisher_path_comb("CHEK2", "EAS")
gnomadv4fisher_path_comb("CHEK2", "SAS")

#PALB2
gnomadv4fisher_path_comb("PALB2", "AFR")
gnomadv4fisher_path_comb("PALB2", "AMR")
gnomadv4fisher_path_comb("PALB2", "EAS")
gnomadv4fisher_path_comb("PALB2", "SAS")

#RAD51C
gnomadv4fisher_path_comb("RAD51C", "AFR")
gnomadv4fisher_path_comb("RAD51C", "AMR")
gnomadv4fisher_path_comb("RAD51C", "EAS")
gnomadv4fisher_path_comb("RAD51C", "SAS")

#RAD51D
gnomadv4fisher_path_comb("RAD51D", "AFR")
gnomadv4fisher_path_comb("RAD51D", "AMR")
gnomadv4fisher_path_comb("RAD51D", "EAS")
gnomadv4fisher_path_comb("RAD51D", "SAS")

#TP53
gnomadv4fisher_path_comb("TP53", "AFR")
gnomadv4fisher_path_comb("TP53", "AMR")
gnomadv4fisher_path_comb("TP53", "EAS")
gnomadv4fisher_path_comb("TP53", "SAS")

#All genes
gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(pathogenic_variants = sum(pathogenic_variants), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic_variants, in_clinvar) %>% 
  fisher.test()

#Function for calculating fisher exact test comparing unreported between populations for genes in clinvar 
gnomadv4fisher_unr_comb <- function(x,y) {gnomadv4_fisher_combination %>% 
  filter(gene == x, population %in% c(y, "NFE")) %>% 
  mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()}

#ATM
gnomadv4fisher_unr_comb("ATM", "AFR")
gnomadv4fisher_unr_comb("ATM", "AMR")
gnomadv4fisher_unr_comb("ATM", "EAS")
gnomadv4fisher_unr_comb("ATM", "SAS")

#BARD1
gnomadv4fisher_unr_comb("BARD1", "AFR")
gnomadv4fisher_unr_comb("BARD1", "AMR")
gnomadv4fisher_unr_comb("BARD1", "EAS")
gnomadv4fisher_unr_comb("BARD1", "SAS")

#BRCA1
gnomadv4fisher_unr_comb("BRCA1", "AFR")
gnomadv4fisher_unr_comb("BRCA1", "AMR")
gnomadv4fisher_unr_comb("BRCA1", "EAS")
gnomadv4fisher_unr_comb("BRCA1", "SAS")

#BRCA2
gnomadv4fisher_unr_comb("BRCA2", "AFR")
gnomadv4fisher_unr_comb("BRCA2", "AMR")
gnomadv4fisher_unr_comb("BRCA2", "EAS")
gnomadv4fisher_unr_comb("BRCA2", "SAS")

#CHEK2
gnomadv4fisher_unr_comb("CHEK2", "AFR")
gnomadv4fisher_unr_comb("CHEK2", "AMR")
gnomadv4fisher_unr_comb("CHEK2", "EAS")
gnomadv4fisher_unr_comb("CHEK2", "SAS")

#PALB2
gnomadv4fisher_unr_comb("PALB2", "AFR")
gnomadv4fisher_unr_comb("PALB2", "AMR")
gnomadv4fisher_unr_comb("PALB2", "EAS")
gnomadv4fisher_unr_comb("PALB2", "SAS")

#RAD51C
gnomadv4fisher_unr_comb("RAD51C", "AFR")
gnomadv4fisher_unr_comb("RAD51C", "AMR")
gnomadv4fisher_unr_comb("RAD51C", "EAS")
gnomadv4fisher_unr_comb("RAD51C", "SAS")

#RAD51D
gnomadv4fisher_unr_comb("RAD51D", "AFR")
gnomadv4fisher_unr_comb("RAD51D", "AMR")
gnomadv4fisher_unr_comb("RAD51D", "EAS")
gnomadv4fisher_unr_comb("RAD51D", "SAS")

#TP53
gnomadv4fisher_unr_comb("TP53", "AFR")
gnomadv4fisher_unr_comb("TP53", "AMR")
gnomadv4fisher_unr_comb("TP53", "EAS")
gnomadv4fisher_unr_comb("TP53", "SAS")

#All genes
gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomadv4_fisher_combination%>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()
```

Fisher Results v2
```{r}
gnomad_path <- gnomad_count %>% mutate(population = case_when(Population == "African_american" ~ "AFR", Population == "Latino_admixed_american" ~ "AMR", Population == "Non_finnish_european" ~ "NFE", Population == "East_asian" ~ "EAS", Population == "South_asian" ~ "SAS"), .after = "Population") %>% select(-Population) %>% filter(clin_var_clinical_significance == "Pathogenic") %>% group_by(gene) %>% count(population) %>% rename("pathogenic" = "n")
  
gnomad_total <- gnomad_count %>% mutate(population = case_when(Population == "African_american" ~ "AFR", Population == "Latino_admixed_american" ~ "AMR", Population == "Non_finnish_european" ~ "NFE", Population == "East_asian" ~ "EAS", Population == "South_asian" ~ "SAS"), .after = "Population") %>% select(-Population) %>% group_by(gene) %>% count(population) %>% 
    rename("total" = "n")
  
gnomad_fisher <- gnomad_count %>% mutate(population = case_when(Population == "African_american" ~ "AFR", Population == "Latino_admixed_american" ~ "AMR", Population == "Non_finnish_european" ~ "NFE", Population == "East_asian" ~ "EAS", Population == "South_asian" ~ "SAS"), .after = "Population") %>% 
  mutate(clin_var_clinical_significance = case_when(clin_var_clinical_significance == "Blank" ~ "unreported", T ~ clin_var_clinical_significance)) %>% 
  select(-Population) %>% unite(test, gene, population, sep=" ") %>% 
  group_by(test) %>%
  count(clin_var_clinical_significance) %>%
  ungroup() %>%
  complete(test, clin_var_clinical_significance, fill = list(n = 0)) %>% 
  group_by(test) %>% 
  filter(!grepl("unreported", clin_var_clinical_significance, ignore.case = TRUE)) %>%
  summarise(in_clinvar = sum(n, na.rm = TRUE)) %>% 
  separate(test, into = c("gene", "population"), sep = " ") %>% 
  full_join(gnomad_path) %>% full_join(gnomad_total) %>%
  mutate(unreported = (total - in_clinvar)) %>% 
  replace_na(list(pathogenic=0))
  
gnomadfisher_path <- function(x,y) {gnomad_fisher %>% 
    filter(gene == x, population %in% c(y, "NFE")) %>% 
    mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
    arrange(population) %>%
    select(pathogenic, in_clinvar) %>% 
    fisher.test()}
  
#ATM
gnomadfisher_path("atm", "AFR")
gnomadfisher_path("atm", "AMR")
gnomadfisher_path("atm", "EAS")
gnomadfisher_path("atm", "SAS")

#BARD1
gnomadfisher_path("bard1", "AFR")
gnomadfisher_path("bard1", "AMR")
gnomadfisher_path("bard1", "EAS")
gnomadfisher_path("bard1", "SAS")

#BRCA1
gnomadfisher_path("brca1", "AFR")
gnomadfisher_path("brca1", "AMR")
gnomadfisher_path("brca1", "EAS")
gnomadfisher_path("brca1", "SAS")

#BRCA2
gnomadfisher_path("brca2", "AFR")
gnomadfisher_path("brca2", "AMR")
gnomadfisher_path("brca2", "EAS")
gnomadfisher_path("brca2", "SAS")

#CHEK2
gnomadfisher_path("chek2", "AFR")
gnomadfisher_path("chek2", "AMR")
gnomadfisher_path("chek2", "EAS")
gnomadfisher_path("chek2", "SAS")

#PALB2
gnomadfisher_path("palb2", "AFR")
gnomadfisher_path("palb2", "AMR")
gnomadfisher_path("palb2", "EAS")
gnomadfisher_path("palb2", "SAS")

#RAD51C
gnomadfisher_path("rad51c", "AFR")
gnomadfisher_path("rad51c", "AMR")
gnomadfisher_path("rad51c", "EAS")
gnomadfisher_path("rad51c", "SAS")

#RAD51D
gnomadfisher_path("rad51d", "AFR")
gnomadfisher_path("rad51d", "AMR")
gnomadfisher_path("rad51d", "EAS")
gnomadfisher_path("rad51d", "SAS")
  
#TP53
gnomadfisher_path("tp53", "AFR")
gnomadfisher_path("tp53", "AMR")
gnomadfisher_path("tp53", "EAS")
gnomadfisher_path("tp53", "SAS")

#All genes
gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic = sum(pathogenic), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>%
  select(pathogenic, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic = sum(pathogenic), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>%
  select(pathogenic, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic = sum(pathogenic), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(pathogenic, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(pathogenic = sum(pathogenic), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>%
  select(pathogenic, in_clinvar) %>% 
  fisher.test()

#Function for calculating fisher exact test comparing unreported between populations for genes in clinvar 
gnomadfisher_unr <- function(x,y) {gnomad_fisher %>% 
    filter(gene == x, population %in% c(y, "NFE")) %>% 
    mutate(population = factor(population, levels = c(y, "NFE"))) %>% 
    arrange(population) %>%
    select(unreported, in_clinvar) %>% 
    fisher.test()}

#ATM
gnomadfisher_unr("atm", "AFR")
gnomadfisher_unr("atm", "AMR")
gnomadfisher_unr("atm", "EAS")
gnomadfisher_unr("atm", "SAS")

#BARD1
gnomadfisher_unr("bard1", "AFR")
gnomadfisher_unr("bard1", "AMR")
gnomadfisher_unr("bard1", "EAS")
gnomadfisher_unr("bard1", "SAS")

#BRCA1
gnomadfisher_unr("brca1", "AFR")
gnomadfisher_unr("brca1", "AMR")
gnomadfisher_unr("brca1", "EAS")
gnomadfisher_unr("brca1", "SAS")

#BRCA2
gnomadfisher_unr("brca2", "AFR")
gnomadfisher_unr("brca2", "AMR")
gnomadfisher_unr("brca2", "EAS")
gnomadfisher_unr("brca2", "SAS")

#CHEK2
gnomadfisher_unr("chek2", "AFR")
gnomadfisher_unr("chek2", "AMR")
gnomadfisher_unr("chek2", "EAS")
gnomadfisher_unr("chek2", "SAS")

#PALB2
gnomadfisher_unr("palb2", "AFR")
gnomadfisher_unr("palb2", "AMR")
gnomadfisher_unr("palb2", "EAS")
gnomadfisher_unr("palb2", "SAS")

#RAD51C
gnomadfisher_unr("rad51c", "AFR")
gnomadfisher_unr("rad51c", "AMR")
gnomadfisher_unr("rad51c", "EAS")
gnomadfisher_unr("rad51c", "SAS")

#RAD51D
gnomadfisher_unr("rad51d", "AFR")
gnomadfisher_unr("rad51d", "AMR")
gnomadfisher_unr("rad51d", "EAS")
gnomadfisher_unr("rad51d", "SAS")

#TP53
gnomadfisher_unr("tp53", "AFR")
gnomadfisher_unr("tp53", "AMR")
gnomadfisher_unr("tp53", "EAS")
gnomadfisher_unr("tp53", "SAS")

#All genes
gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AFR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AFR", "NFE"))) %>% 
  arrange(population) %>%
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("AMR", "NFE")) %>% 
  mutate(population = factor(population, levels = c("AMR", "NFE"))) %>% 
  arrange(population) %>%
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("EAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("EAS", "NFE"))) %>% 
  arrange(population) %>% 
  select(unreported, in_clinvar) %>% 
  fisher.test()

gnomad_fisher %>% 
  group_by(population) %>% 
  summarise(unreported = sum(unreported), in_clinvar = sum(in_clinvar)) %>% 
  filter(population %in% c("SAS", "NFE")) %>% 
  mutate(population = factor(population, levels = c("SAS", "NFE"))) %>% 
  arrange(population) %>%
  select(unreported, in_clinvar) %>% 
  fisher.test()
```

Tables for paper
```{r}
gnomadv4_count %>% group_by(gene) %>% count(population) %>% pivot_wider(names_from = population, values_from = n) %>% ungroup() %>% mutate(Total = rowSums(select(.,-gene))) %>% bind_rows(summarise(.,across(where(is.numeric), sum, na.rm=T)) %>% mutate(gene="Total")) %>% write.csv(here("output/v4/csv files/population.csv"), row.names = F) #Number of variants per population in each gene

gnomadv4_count %>% group_by(gene) %>% filter(clin_var_clinical_significance=="Unreported") %>% count(population) %>% pivot_wider(names_from = population, values_from = n) %>% ungroup() %>% mutate(Total = rowSums(select(.,-gene))) %>% bind_rows(summarise(.,across(where(is.numeric), sum, na.rm=T)) %>% mutate(gene="Total")) %>% write.csv(here("output/v4/csv files/population_unr.csv"), row.names = F) #Number of variants per population in each gene that is unreported in ClinVar

gnomadv4_count %>% group_by(gene) %>% count(clin_var_clinical_significance) %>% pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>% ungroup() %>% mutate(Total = rowSums(select(.,-gene))) %>% bind_rows(summarise(.,across(where(is.numeric), sum, na.rm=T)) %>% mutate(gene="Total")) %>% write.csv(here("output/v4/csv files/clinvar.csv"), row.names = F) # Number of variants per ClinVar annotation in each gene 

gnomadv4 %>% 
  rename("AFR" = "allele_count_african_african_american", "AMR" = "allele_count_admixed_american", "EAS" = "allele_count_east_asian", "NFE" = "allele_count_european_non_finnish", "SAS" = "allele_count_south_asian") %>% 
  pivot_longer(names_to = "population", values_to = "values", cols = c("AFR", "AMR", "NFE", "EAS", "SAS")) %>% 
  select(gnom_ad_id, gene, population, values) %>% filter(values!=0) %>% group_by(population) %>% count(gene) %>% pivot_wider(names_from = "population", values_from = "n") %>% mutate(Total = rowSums(select(.,-gene))) %>% ungroup() %>% bind_rows(summarise(.,across(where(is.numeric), sum, na.rm=T)) %>% mutate(gene="Total")) %>% write_csv(here("output/v4/csv files/total_populations.csv")) #Total number of variants in GnomAD in each gene that appear in the 5 populations

gnomadv4_count %>% 
  unite(test, gene, clin_var_clinical_significance, sep = " ") %>% 
  group_by(test) %>%
  count(count_range) %>% 
  complete(count_range, fill = list(n=0)) %>% 
  mutate(n= case_when(is.na(n)~ 0, TRUE~n)) %>% 
  mutate(normalised = round((n / sum(n)) * 100, 2)) %>% 
  select(test, count_range, normalised) %>% 
  pivot_wider(names_from = count_range, values_from = normalised) %>% 
  select(test, "0-1", "2-3", "4-10", "11-100", "100+") %>% 
  replace_na(list("0-1"=0, "2-3"=0, "4-10"=0, "11-100"=0,"100+"=0)) %>% 
  separate(test, into = c("gene", "clinvar_significance"), sep = " ") %>% 
  write_csv(here("output/v4/csv files/allele_count.csv")) #Percentage of variants in each gene and clinvar significance in each allele count range

pca_gnomadv4_modified_rank %>% 
  count(gene) %>% 
  write.csv(here("output/v4/csv files/pca_breakdown.csv")) #Gene breakdown of PCA

gnomadv4_count %>%
  filter(clin_var_clinical_significance=="Unreported") %>% 
  group_by(gene) %>%
  count(vep_annotation) %>%
  mutate(percent = round((n/sum(n))*100,2),
         vep_annotation = case_when(vep_annotation == "frameshift_variant" ~ "Frameshift Variant",
                                    vep_annotation == "inframe_variant" ~ "Inframe Variant",
                                    vep_annotation == "missense_variant" ~ "Missense Variant",
                                    vep_annotation == "splice_site_variant" ~ "Splice Site Variant",
                                    vep_annotation == "start_lost" ~ "Start Lost",
                                    vep_annotation == "stop_lost" ~ "Stop Lost",
                                    vep_annotation == "stop_gained" ~ "Stop Gained")) #Fig1D breakdown
```
