gnomad 2022 - Line graph of percentage of annotation of population unique variants.

```{r}
library(tidyverse)
library(here)
library(janitor)
library(visdat)
library(skimr)
library(rstatix)
library(dplyr)
library(knitr)
```
Joining all data of genes from 2022 into 1 file
```{r}
atm_2["chromosome"] <- "atm"
bard1_2["chromosome"] <- "bard1"
brca1_2["chromosome"] <- "brca1"
brca2_2["chromosome"] <- "brca2"
chek2_2["chromosome"] <- "chek2"
palb2_2["chromosome"] <- "palb2"
rad51c_2["chromosome"] <- "rad51c"
rad51d_2["chromosome"] <- "rad51d"
tp53_2["chromosome"] <- "tp53"

full_join(atm_2, brca1_2) %>% full_join(bard1_2) %>% full_join(brca2_2) %>% full_join(chek2_2) %>% full_join(palb2_2) %>% full_join(rad51c_2) %>% full_join(rad51d_2) %>% full_join(tp53_2) -> gnomad_2

Pathogenic <- c("Pathogenic", "Pathogenic/Likely pathogenic", "Likely pathogenic")
Benign <- c("Benign", "Benign/Likely benign", "Likely benign")
Uncertain <- c("Uncertain significance", "Conflicting interpretations of pathogenicity", "Conflicting interpretations of pathogenicity; risk factor")
Splice <- c("splice_donor_variant", "splice_acceptor_variant")
Inframe <- c("inframe_deletion", "inframe_insertion")
Stop <- c("stop_gained", "stop_lost")
```

Creating separate data on basis of population
```{r}
gnomad_2 %>% filter(allele_count_african_african_american!=0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>%
filter(!is.na(clin_var_clinical_significance)) %>% count(chromosome) %>% rename(African_2022=n) ->gnomad_2_afro

gnomad_2 %>% filter(allele_count_african_african_american!=0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>% count(chromosome) %>% rename(African_2022_total=n) ->gnomad_2_afro_total

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american!=0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>% 
filter(!is.na(clin_var_clinical_significance)) %>% count(chromosome) %>% rename(Latino_2022=n)-> gnomad_2_latino

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american!=0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>% 
count(chromosome) %>% rename(Latino_2022_total=n)-> gnomad_2_latino_total

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish!=0 , allele_count_south_asian==0) %>% 
filter(!is.na(clin_var_clinical_significance)) %>% count(chromosome)%>% rename(Europe_2022=n)-> gnomad_2_europe

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish!=0 , allele_count_south_asian==0) %>% 
count(chromosome)%>% rename(Europe_2022_total=n)-> gnomad_2_europe_total

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian!=0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>% 
filter(!is.na(clin_var_clinical_significance)) %>% count(chromosome)%>% rename(East_asia_2022=n)-> gnomad_2_east_asia

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian!=0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0) %>% 
count(chromosome)%>% rename(East_asia_2022_total=n)-> gnomad_2_east_asia_total

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian!=0) %>% 
filter(!is.na(clin_var_clinical_significance)) %>% count(chromosome)%>% rename(South_asia_2022=n) -> gnomad_2_south_asia

gnomad_2 %>% filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian!=0) %>% 
count(chromosome)%>% rename(South_asia_2022_total=n) -> gnomad_2_south_asia_total
```

Number of annotated population unique variants
```{r}
full_join(gnomad_2_afro,gnomad_2_latino) %>% full_join(gnomad_2_europe) %>% full_join(gnomad_2_east_asia) %>% full_join(gnomad_2_south_asia) %>% pivot_longer(cols=c(African_2022, Latino_2022, Europe_2022, East_asia_2022, South_asia_2022), names_to = "Population", values_to = "Values") %>% ggplot(aes(x=chromosome,y=Values,fill=Population)) + geom_col()+ facet_wrap(~Population) + theme(axis.text.x = element_text(angle = 35, hjust=1))

full_join(gnomad_2_afro,gnomad_2_latino) %>% full_join(gnomad_2_europe) %>% full_join(gnomad_2_east_asia) %>% full_join(gnomad_2_south_asia) %>% pivot_longer(cols=c(African_2022, Latino_2022, Europe_2022, East_asia_2022, South_asia_2022), names_to = "Population", values_to = "Values") %>% ggplot(aes(x=Population, y=Values, shape=Population)) + geom_point() + facet_wrap(~chromosome) + theme(axis.text.x = element_text(angle = 35, hjust=1)) + geom_text(aes(label=Values), vjust=-1) + ylim(0, 1000) + scale_x_discrete(limits = c("Europe_2022", "African_2022", "Latino_2022", "East_asia_2022", "South_asia_2022")) + labs(y="Number of population unique variants annotated")
```

Percentage of annotation of population unique variants.
```{r}
full_join(gnomad_2_afro,gnomad_2_latino) %>% full_join(gnomad_2_europe) %>% full_join(gnomad_2_east_asia) %>% full_join(gnomad_2_south_asia) %>% full_join(gnomad_2_afro_total) %>% full_join(gnomad_2_europe_total) %>% full_join(gnomad_2_east_asia_total) %>% full_join(gnomad_2_latino_total) %>% full_join(gnomad_2_south_asia_total) %>% 
  mutate("% annotated African 2022"=round((African_2022/African_2022_total)*100,2), 
         "% annotated Latino 2022"=round((Latino_2022/Latino_2022_total)*100,2), 
         "% annotated Europe 2022"=round((Europe_2022/Europe_2022_total)*100,2), 
         "% annotated East asia 2022"=round((East_asia_2022/East_asia_2022_total)*100,2),
         "% annotated South asia 2022"=round((South_asia_2022/South_asia_2022_total)*100,2)) %>%
  select(-c(Europe_2022, East_asia_2022, African_2022, South_asia_2022, Latino_2022)) %>% 
  rename(Europe_2022="% annotated Europe 2022",East_asia_2022="% annotated East asia 2022",Latino_2022="% annotated Latino 2022",South_asia_2022="% annotated South asia 2022",African_2022="% annotated African 2022") %>%
  pivot_longer(cols=c(African_2022, Latino_2022, Europe_2022, East_asia_2022, South_asia_2022), names_to = "Population_2022", values_to = "Values_2022") %>% 
  select(chromosome, Population_2022, Values_2022) %>% 
  ggplot(aes(x=Population_2022,y=Values_2022, shape=Population_2022)) + geom_point() + geom_line(aes(group = chromosome), color = "black", linetype=6)+ facet_wrap(~chromosome) + theme(axis.text.x = element_text(angle = 35, hjust=1)) + ylim(0, 100) + scale_x_discrete(limits = c("Europe_2022", "African_2022", "Latino_2022", "East_asia_2022", "South_asia_2022")) + labs(title= "Line graph of each population showing the percentage of 
clinical annotation of all variants", y= "Percentage annotated")

ggsave(here("output/variants_2_line.png"), width = 8, height = 6, dpi = 300)
```