dbNSFP - Missense variants of all genes - Prediction and Ranking softwares refined.

```{r}
library(tidyverse)
library(here)
library(janitor)
library(visdat)
library(skimr)
library(rstatix)
library(dplyr)
library(knitr)

Pathogenic <- c("Pathogenic", "Pathogenic/Likely pathogenic", "Likely pathogenic")
Benign <- c("Benign", "Benign/Likely benign", "Likely benign")
Uncertain <- c("Uncertain significance", "Conflicting interpretations of pathogenicity", "Conflicting interpretations of pathogenicity; risk factor")
Splice <- c("splice_donor_variant", "splice_acceptor_variant")
Inframe <- c("inframe_deletion", "inframe_insertion")
Stop <- c("stop_gained", "stop_lost")

dbnsfp <- function() {gnomad %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant")}
```

dbNSFP
ATM input for dbNSFP
```{r}
dbnsfp() %>% 
  filter(chromosome==11) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> atmDbnsfp_path

atmDbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/atm_path.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==11) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> atmDbnsfp_ben

atmDbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/atm_ben.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==11) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> atmDbnsfp_uncertain

atmDbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/atm_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==11) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> atmDbnsfp_blank

atmDbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/atm_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
ATM output reading and removing duplicates
```{r}
atm_ben <- (read.table(file = here("output/dbnsfp/Worked/atm_ben.txt"), header = T))
atm_blank <- read.table(file = here("output/dbnsfp/Worked/atm_blank.txt"), header = T)
atm_path <- read.table(file = here("output/dbnsfp/Worked/atm_path.txt"), header = T)
atm_uncertain <- read.table(file = here("output/dbnsfp/Worked/atm_uncertain.txt"), header = T)

atm_ben <- atm_ben[!duplicated(atm_ben$rs_dbSNP),]
atm_blank <- atm_blank[!duplicated(atm_blank$rs_dbSNP),]
atm_path <- atm_path[!duplicated(atm_path$rs_dbSNP),]
atm_uncertain <- atm_uncertain[!duplicated(atm_uncertain$rs_dbSNP),]
```
Ben - 27/28, Path - 17/17, Blank - 233/234, Uncertain - 1164/1251

BARD1 input for dbNSFP
```{r}
dbnsfp() %>% 
  filter(chromosome==2) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> bard1Dbnsfp_path

bard1Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/bard1_path.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==2) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> bard1Dbnsfp_ben

bard1Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/bard1_ben.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==2) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> bard1Dbnsfp_uncertain

bard1Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/bard1_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

dbnsfp() %>% 
  filter(chromosome==2) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> bard1Dbnsfp_blank

bard1Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/bard1_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
BARD1 output reading and removing duplicates
```{r}
bard1_ben <- read.table(file = here("output/dbnsfp/Worked/bard1_ben.txt"), header = T) 
bard1_blank <- read.table(file = here("output/dbnsfp/Worked/bard1_blank.txt"), header = T)
bard1_uncertain <- read.table(file = here("output/dbnsfp/Worked/bard1_uncertain.txt"), header = T)

bard1_ben <- bard1_ben[!duplicated(bard1_ben$pos),]
bard1_blank <- bard1_blank[!duplicated(bard1_blank$pos),]
bard1_uncertain<- bard1_uncertain[!duplicated(bard1_uncertain$pos),]
```
Ben - 13/19 Path - 0/0  Blank - 60/62  Uncertain - 377/407

BRCA1 input for dbNSFP
```{r}
brca1 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca1Dbnsfp_path

brca1Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca1_path.txt"), sep = "\t", row.names = F, col.names = F)

brca1 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca1Dbnsfp_ben

brca1Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca1_ben.txt"), sep = "\t", row.names = F, col.names = F)

brca1 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca1Dbnsfp_uncertain

brca1Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca1_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

brca1 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca1Dbnsfp_blank

brca1Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/brca1_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
BRCA1 output reading and removing duplicates
```{r}
brca1_ben <- read.table(file = here("output/dbnsfp/Worked/brca1_ben.txt"), header = T)
brca1_blank <- read.table(file = here("output/dbnsfp/Worked/brca1_blank.txt"), header = T)
brca1_path <- read.table(file = here("output/dbnsfp/Worked/brca1_path.txt"), header = T)
brca1_uncertain <- read.table(file = here("output/dbnsfp/Worked/brca1_uncertain.txt"), header = T)

brca1_ben <- brca1_ben[!duplicated(brca1_ben$pos),]
brca1_blank <- brca1_blank[!duplicated(brca1_blank$pos),]
brca1_path<- brca1_path[!duplicated(brca1_path$pos),]
brca1_uncertain<- brca1_uncertain[!duplicated(brca1_uncertain$pos),]
```
Ben - 152/156 Path - 17/17  Blank - 136/137 Uncertain - 631/656

BRCA2 input for dbNSFP
```{r}
brca2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==13) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca2Dbnsfp_path

brca2Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca2_path.txt"), sep = "\t", row.names = F, col.names = F)

brca2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==13) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca2Dbnsfp_ben

brca2Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca2_ben.txt"), sep = "\t", row.names = F, col.names = F)

brca2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==13) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca2Dbnsfp_uncertain

brca2Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/brca2_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

brca2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==13) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> brca2Dbnsfp_blank

brca2Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/brca2_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
BRCA2 output reading and removing duplicates
```{r}
brca2_ben <- read.table(file = here("output/dbnsfp/Worked/brca2_ben.txt"), header = T) 
brca2_blank <- read.table(file = here("output/dbnsfp/Worked/brca2_blank.txt"), header = T)
brca2_path <- read.table(file = here("output/dbnsfp/Worked/brca2_path.txt"), header = T)
brca2_uncertain <- read.table(file = here("output/dbnsfp/Worked/brca2_uncertain.txt"), header = T)

brca2_ben <- brca2_ben[!duplicated(brca2_ben$pos),]
brca2_blank <- brca2_blank[!duplicated(brca2_blank$pos),]
brca2_path<- brca2_path[!duplicated(brca2_path$pos),]
brca2_uncertain<- brca2_uncertain[!duplicated(brca2_uncertain$pos),]
```
Ben - 160/161 Path - 12/12 Blank - 265/266 Uncertain - 1420/1508

CHEK2 input for dbNSFP
```{r}
chek2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==22) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> chek2Dbnsfp_path

chek2Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/chek2_path.txt"), sep = "\t", row.names = F, col.names = F)

chek2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==22) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> chek2Dbnsfp_ben

chek2Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/chek2_ben.txt"), sep = "\t", row.names = F, col.names = F)

chek2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==22) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> chek2Dbnsfp_uncertain

chek2Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/chek2_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

chek2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==22) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> chek2Dbnsfp_blank

chek2Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/chek2_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
BRCA2 output reading and removing duplicates
```{r}
chek2_ben <- read.table(file = here("output/dbnsfp/Worked/chek2_ben.txt"), header = T) 
chek2_blank <- read.table(file = here("output/dbnsfp/Worked/chek2_blank.txt"), header = T)
chek2_path <- read.table(file = here("output/dbnsfp/Worked/chek2_path.txt"), header = T)
chek2_uncertain <- read.table(file = here("output/dbnsfp/Worked/chek2_uncertain.txt"), header = T)

chek2_ben <- chek2_ben[!duplicated(chek2_ben$pos),]
chek2_blank <- chek2_blank[!duplicated(chek2_blank$pos),]
chek2_path<- chek2_path[!duplicated(chek2_path$pos),]
chek2_uncertain<- chek2_uncertain[!duplicated(chek2_uncertain$pos),]
```
Ben - 3/6 Path - 2/2  Blank - 42/47 Uncertain - 269/303

PALB2 input for dbNSFP
```{r}
palb2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==16) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> palb2Dbnsfp_path

palb2Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/palb2_path.txt"), sep = "\t", row.names = F, col.names = F)

palb2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==16) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> palb2Dbnsfp_ben

palb2Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/palb2_ben.txt"), sep = "\t", row.names = F, col.names = F)

palb2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==16) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> palb2Dbnsfp_uncertain

palb2Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/palb2_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

palb2 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==16) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> palb2Dbnsfp_blank

palb2Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/palb2_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
PALB2 output reading and removing duplicates
```{r}
palb2_ben <- read.table(file = here("output/dbnsfp/Worked/palb2_ben.txt"), header = T) 
palb2_blank <- read.table(file = here("output/dbnsfp/Worked/palb2_blank.txt"), header = T)
palb2_uncertain <- read.table(file = here("output/dbnsfp/Worked/palb2_uncertain.txt"), header = T)

palb2_ben <- palb2_ben[!duplicated(palb2_ben$pos),]
palb2_blank <- palb2_blank[!duplicated(palb2_blank$pos),]
palb2_uncertain<- palb2_uncertain[!duplicated(palb2_uncertain$pos),]
```
Ben - 11/11 Path - 0/0  Blank - 76/76 Uncertain - 496/532

RAD51C input for dbNSFP
```{r}
rad51c %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51cDbnsfp_path

rad51cDbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51c_path.txt"), sep = "\t", row.names = F, col.names = F)

rad51c %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51cDbnsfp_ben

rad51cDbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51c_ben.txt"), sep = "\t", row.names = F, col.names = F)

rad51c %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51cDbnsfp_uncertain

rad51cDbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51c_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

rad51c %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51cDbnsfp_blank

rad51cDbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/rad51c_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
RAD51C output reading and removing duplicates
```{r}
rad51c_ben <- read.table(file = here("output/dbnsfp/Worked/rad51c_ben.txt"), header = T) 
rad51c_blank <- read.table(file = here("output/dbnsfp/Worked/rad51c_blank.txt"), header = T)
rad51c_path <- read.table(file = here("output/dbnsfp/Worked/rad51c_path.txt"), header = T)
rad51c_uncertain <- read.table(file = here("output/dbnsfp/Worked/rad51c_uncertain.txt"), header = T)

rad51c_ben <- rad51c_ben[!duplicated(rad51c_ben$pos),]
rad51c_blank <- rad51c_blank[!duplicated(rad51c_blank$pos),]
rad51c_path<- rad51c_path[!duplicated(rad51c_path$pos),]
rad51c_uncertain<- rad51c_uncertain[!duplicated(rad51c_uncertain$pos),]
```
Ben - 5/5 Path - 2/2  Blank - 46/46 Uncertain - 169/183

RAD51D input for dbNSFP
```{r}
rad51d %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51dDbnsfp_path

rad51dDbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51d_path.txt"), sep = "\t", row.names = F, col.names = F)

rad51d %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51dDbnsfp_ben

rad51dDbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51d_ben.txt"), sep = "\t", row.names = F, col.names = F)

rad51d %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51dDbnsfp_uncertain

rad51dDbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/rad51d_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

rad51d %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> rad51dDbnsfp_blank

rad51dDbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/rad51d_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
RAD51D output reading and removing duplicates
```{r}
rad51d_ben <- read.table(file = here("output/dbnsfp/Worked/rad51d_ben.txt"), header = T) 
rad51d_blank <- read.table(file = here("output/dbnsfp/Worked/rad51d_blank.txt"), header = T)
rad51d_uncertain <- read.table(file = here("output/dbnsfp/Worked/rad51d_uncertain.txt"), header = T)

rad51d_ben <- rad51d_ben[!duplicated(rad51d_ben$pos),]
rad51d_blank <- rad51d_blank[!duplicated(rad51d_blank$pos),]
rad51d_uncertain<- rad51d_uncertain[!duplicated(rad51d_uncertain$pos),]
```
Ben - 13/14 Path - 0/0  Blank - 67/79 Uncertain - 155/166

TP53 input for dbNSFP
```{r}
tp53 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Pathogenic") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> tp53Dbnsfp_path

tp53Dbnsfp_path$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/tp53_path.txt"), sep = "\t", row.names = F, col.names = F)

tp53 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Benign") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> tp53Dbnsfp_ben

tp53Dbnsfp_ben$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/tp53_ben.txt"), sep = "\t", row.names = F, col.names = F)

tp53 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(clin_var_clinical_significance=="Uncertain") %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> tp53Dbnsfp_uncertain

tp53Dbnsfp_uncertain$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/tp53_uncertain.txt"), sep = "\t", row.names = F, col.names = F)

tp53 %>%  
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% filter(vep_annotation=="missense_variant") %>%
  filter(chromosome==17) %>% 
  filter(is.na(clin_var_clinical_significance)) %>% 
  unite(dbnsfp,chromosome, position, reference, alternate, sep =' ') -> tp53Dbnsfp_blank

tp53Dbnsfp_blank$dbnsfp -> dbNsfp

write.table(dbNsfp, file=here("output/dbnsfp/tp53_blank.txt"), sep = "\t", row.names = F, col.names = F)
```
TP53 output reading and removing duplicates
```{r}
tp53_ben <- read.table(file = here("output/dbnsfp/Worked/tp53_ben.txt"), header = T) 
tp53_blank <- read.table(file = here("output/dbnsfp/Worked/tp53_blank.txt"), header = T)
tp53_path <- read.table(file = here("output/dbnsfp/Worked/tp53_path.txt"), header = T)
tp53_uncertain <- read.table(file = here("output/dbnsfp/Worked/tp53_uncertain.txt"), header = T)

tp53_ben <- tp53_ben[!duplicated(tp53_ben$pos),]
tp53_blank <- tp53_blank[!duplicated(tp53_blank$pos),]
tp53_path<- tp53_path[!duplicated(tp53_path$pos),]
tp53_uncertain<- tp53_uncertain[!duplicated(tp53_uncertain$pos),]
```
Ben - 55/59 Path - 21/24  Blank - 35/37 Uncertain - 105/113

Joining all the variants
```{r}
atm_ben <- atm_ben %>% mutate_all(as.character)
atm_path <- atm_path%>% mutate_all(as.character)
atm_blank <- atm_blank%>% mutate_all(as.character)
atm_uncertain <- atm_uncertain%>% mutate_all(as.character)
bard1_ben <- bard1_ben %>% mutate_all(as.character)

bard1_blank <- bard1_blank%>% mutate_all(as.character)
bard1_uncertain <- bard1_uncertain%>% mutate_all(as.character)
brca1_ben <- brca1_ben %>% mutate_all(as.character)
brca1_path <- brca1_path%>% mutate_all(as.character)
brca1_blank <- brca1_blank%>% mutate_all(as.character)
brca1_uncertain <- brca1_uncertain%>% mutate_all(as.character)
brca2_ben <- brca2_ben %>% mutate_all(as.character)
brca2_path <- brca2_path%>% mutate_all(as.character)
brca2_blank <- brca2_blank%>% mutate_all(as.character)
brca2_uncertain <- brca2_uncertain%>% mutate_all(as.character)
chek2_ben <- chek2_ben %>% mutate_all(as.character)
chek2_path <- chek2_path%>% mutate_all(as.character)
chek2_blank <- chek2_blank%>% mutate_all(as.character)
chek2_uncertain <- chek2_uncertain%>% mutate_all(as.character)
palb2_ben <- palb2_ben %>% mutate_all(as.character)

palb2_blank <- palb2_blank%>% mutate_all(as.character)
palb2_uncertain <- palb2_uncertain%>% mutate_all(as.character)
rad51c_ben <- rad51c_ben %>% mutate_all(as.character)
rad51c_path <- rad51c_path%>% mutate_all(as.character)
rad51c_blank <- rad51c_blank%>% mutate_all(as.character)
rad51c_uncertain <- rad51c_uncertain%>% mutate_all(as.character)
rad51d_ben <- rad51d_ben %>% mutate_all(as.character)

rad51d_blank <- rad51d_blank%>% mutate_all(as.character)
rad51d_uncertain <- rad51d_uncertain%>% mutate_all(as.character)
tp53_ben <- tp53_ben %>% mutate_all(as.character)
tp53_path <- tp53_path%>% mutate_all(as.character)
tp53_blank <- tp53_blank%>% mutate_all(as.character)
tp53_uncertain <- tp53_uncertain%>% mutate_all(as.character)


prediction <- full_join(atm_ben, atm_path) %>% full_join(atm_blank) %>% full_join(atm_uncertain) %>% full_join(bard1_ben) %>% full_join(bard1_blank) %>% full_join(bard1_uncertain) %>% full_join(brca1_ben) %>% full_join(brca1_path) %>% full_join(brca1_blank) %>% full_join(brca1_uncertain) %>% full_join(brca2_ben) %>% full_join(brca2_path) %>% full_join(brca2_blank) %>% full_join(brca2_uncertain) %>% full_join(chek2_ben) %>% full_join(chek2_path) %>% full_join(chek2_uncertain) %>% full_join(chek2_blank) %>% full_join(palb2_ben) %>% full_join(palb2_blank) %>% full_join(palb2_uncertain) %>% full_join(rad51c_ben) %>% full_join(rad51c_blank) %>% full_join(rad51c_path) %>% full_join(rad51c_uncertain) %>% full_join(rad51d_ben) %>% full_join(rad51d_uncertain) %>% full_join(rad51d_blank) %>% full_join(tp53_ben) %>% full_join(tp53_path) %>% full_join(tp53_uncertain) %>% full_join(tp53_blank) %>% mutate(genename= case_when(genename=="RAD51L3-RFFL" ~ "RAD51D", TRUE~genename))

dbNSFP_prediction <- prediction %>%
  mutate(MetaSVM_pred=ifelse(MetaSVM_rankscore>"0.82257","D",
                             ifelse(MetaSVM_rankscore==".", "B", "T")),
         MetaLR_pred=ifelse(MetaLR_rankscore>"0.81101","D",
                            ifelse(MetaLR_rankscore==".", "B", "T")),
         Polyphen2_HDIV_pred=ifelse(Polyphen2_HDIV_rankscore >"0.38028", "D", 
                                    ifelse(Polyphen2_HDIV_rankscore==".", "B", "T")),
         SIFT_pred=ifelse(SIFT_converted_rankscore>"0.39575","D",
                          ifelse(SIFT_converted_rankscore==".", "B", "T")),
         MutationAssessor_pred=ifelse(MutationAssessor_rankscore>"0.9307", "H",
                               ifelse(MutationAssessor_rankscore>"0.52043", "M",
                               ifelse(MutationAssessor_rankscore>"0.19675","L",
                                      ifelse(MutationAssessor_rankscore==".", "B", "N")))),
         FATHMM_pred=ifelse(FATHMM_converted_rankscore>="0.81332", "D", 
                            ifelse(FATHMM_converted_rankscore==".", "B", "T"))) %>% 
  unite(dbNSFP,chr, hg19_pos, ref, alt, sep = " ") %>%
  select(genename, dbNSFP, MetaSVM_pred, MetaLR_pred, Polyphen2_HDIV_pred,  FATHMM_pred, SIFT_pred, MutationAssessor_pred) %>% 
  mutate(count = rowSums(select(., MetaSVM_pred, MetaLR_pred, Polyphen2_HDIV_pred,  FATHMM_pred, SIFT_pred, MutationAssessor_pred) == "D" |
                             select(., MetaSVM_pred, MetaLR_pred, Polyphen2_HDIV_pred,  FATHMM_pred, SIFT_pred, MutationAssessor_pred) == "H" |
                             select(.,MetaSVM_pred, MetaLR_pred, Polyphen2_HDIV_pred,  FATHMM_pred, SIFT_pred, MutationAssessor_pred) == "M")) %>% 
  mutate(blank_count = rowSums(select(., MetaSVM_pred, MetaLR_pred, Polyphen2_HDIV_pred,  FATHMM_pred, SIFT_pred, MutationAssessor_pred) == "B"))

write.csv(dbNSFP_prediction, file = here("output/csv files/predictions.csv"), row.names =F)

dbNSFP_rankscores <- 
  prediction %>%
  unite(dbNSFP,chr, hg19_pos, ref, alt, sep = " ") %>% 
  select(genename, dbNSFP, SIFT_converted_rankscore, Polyphen2_HDIV_rankscore, Polyphen2_HVAR_rankscore,MetaLR_rankscore, MetaSVM_rankscore, MutationAssessor_rankscore, FATHMM_converted_rankscore, VEST4_rankscore, REVEL_rankscore, CADD_raw_rankscore, CADD_raw_rankscore_hg19, GERP_RS_rankscore)
  
write.csv(dbNSFP_rankscores, file = here("output/csv files/rankscores.csv"), row.names =F)
```
Filtering for only unique variants from populations
```{r}
gnomad_unique %>% 
  rename(gene=chromosome) %>% 
  mutate(chromosome = case_when(gene=="atm" ~ 11,
                                gene=="bard1" ~ 2,
                                gene=="brca1" ~ 17,
                                gene=="brca2" ~ 13,
                                gene=="chek2" ~ 22,
                                gene=="palb2" ~ 16,
                                gene=="rad51c" ~ 17,
                                gene=="rad51d" ~ 17,
                                gene=="tp53" ~ 17),.after = gene) %>%
  unite(variant_id, chromosome, position, reference, alternate, sep = " ") %>% 
  pull(variant_id) -> unique_variants

unique_prediction <- dbNSFP_prediction %>%
  filter(dbNSFP %in% unique_variants)

unique_rankscores <- dbNSFP_rankscores %>%
  filter(dbNSFP %in% unique_variants)
```
Adding population and clin_var significance
```{r}
unique_prediction %>% pull(dbNSFP) -> dbNSFP_pred
gnomad_unique %>% 
  rename(gene=chromosome) %>% 
  mutate(chromosome = case_when(gene=="atm" ~ 11,
                                gene=="bard1" ~ 2,
                                gene=="brca1" ~ 17,
                                gene=="brca2" ~ 13,
                                gene=="chek2" ~ 22,
                                gene=="palb2" ~ 16,
                                gene=="rad51c" ~ 17,
                                gene=="rad51d" ~ 17,
                                gene=="tp53" ~ 17),.after = gene) %>% 
  unite(variant_id, chromosome, position, reference, alternate, sep = " ") %>%
  mutate(clin_var_clinical_significance=case_when(is.na(clin_var_clinical_significance)~ "blank",
                                                  TRUE ~ clin_var_clinical_significance)) %>% 
  unite(conditions, Population, clin_var_clinical_significance, sep = " ") -> population_specific

population_specific %>% 
  filter(variant_id %in% dbNSFP_pred) %>% select(variant_id, conditions) -> test

unique_prediction %>% rename(gene=genename, variant_id=dbNSFP) %>% inner_join(test, by="variant_id") %>%  separate(conditions, into = c("Country", "Clin_var_significance"), sep = " ") %>% 
  mutate(Population = case_when(Country=="afr" ~ "African",
                                Country=="euro" ~ "European",
                                Country=="lat" ~ "Latino",
                                Country=="ea" ~ "East_asian",
                                Country=="sa" ~ "South_asian"),.after=Country) %>%
  select(-Country) %>% 
  mutate(Percentage= round((count/7)*100,2)) -> unique_pred_percent

write.csv(unique_pred_percent, file=here("output/csv files/Populations specific prediction.csv"), row.names = F)

unique_rankscores %>% 
  rename(gene=genename, variant_id=dbNSFP) %>% inner_join(test, by="variant_id") %>% 
  separate(conditions, into = c("Country", "Clin_var_significance"), sep = " ") %>% 
  mutate(Population = case_when(Country=="afr" ~ "African",
                                Country=="euro" ~ "European",
                                Country=="lat" ~ "Latino",
                                Country=="ea" ~ "East_asian",
                                Country=="sa" ~ "South_asian"),.after=Country) %>% select(-Country) -> unique_rankscore

write.csv(unique_rankscore, file=here("output/csv files/Populations specific rankscores.csv"), row.names = F)
```

