gnomAD 2023 - ATM - Population specific variant table & Comparison of unique and shared variants within population

```{r}
library(tidyverse)
library(here)
library(janitor)
library(visdat)
library(skimr)
library(rstatix)
library(knitr)
```

Loading data
```{r}
atm <- clean_names(read_csv(here("data/ATM.csv"))) %>% mutate(gene="ATM",.before=chromosome)

atm_african_unique <- function() {atm %>% 
  filter(allele_count_african_african_american!=0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0)}

atm_latino_unique <- function() {atm %>% 
   filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american!=0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0)}

atm_european_unique <- function() {atm %>% 
   filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish!=0 , allele_count_south_asian==0)}

atm_east_asian_unique <- function() {atm %>% 
   filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian!=0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian==0)}

atm_south_asian_unique <- function() {atm %>% 
   filter(allele_count_african_african_american==0 & allele_count_other==0 & allele_count_latino_admixed_american==0 & allele_count_ashkenazi_jewish==0 & allele_count_east_asian==0 & allele_count_european_finnish==0 & allele_count_european_non_finnish==0 , allele_count_south_asian!=0)}

Pathogenic <- c("Pathogenic", "Pathogenic/Likely pathogenic", "Likely pathogenic")
Benign <- c("Benign", "Benign/Likely benign", "Likely benign")
Uncertain <- c("Uncertain significance", "Conflicting interpretations of pathogenicity", "Conflicting interpretations of pathogenicity; risk factor")
Splice <- c("splice_donor_variant", "splice_acceptor_variant")
Inframe <- c("inframe_deletion", "inframe_insertion")
Stop <- c("stop_gained", "stop_lost")
```

AFR
```{r}
atm_african_table <- atm_african_unique() %>% 
  group_by(vep_annotation) %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation)) %>% 
  count(clin_var_clinical_significance) %>% mutate(Total= sum(n)) %>% 
  mutate(Percent_in_ClinVar=round(prop.table(n),4)*100) %>% 
  mutate(Percent_in_ClinVar = ifelse(is.na(clin_var_clinical_significance), 0, Percent_in_ClinVar)) %>%
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>%
  mutate(Benign = ifelse(!"Benign" %in% colnames(.), 0, Benign)) %>%
  replace_na(list("Pathogenic" = 0, "Uncertain" = 0, "Benign" = 0, "NA" = 0)) %>%
  rename(Not_in_ClinVar = "NA") %>%
  mutate(In_Clinvar = sum(Pathogenic, Benign, Uncertain)) %>% 
  summarise("% in ClinVar" = sum(Percent_in_ClinVar), Total=mean(Total), "Benign/Likely Benign"=sum(Benign), "Pathogenic/Likely Pathogenic"=sum(Pathogenic), Uncertain=sum(Uncertain), "Not in Clinvar"=sum(Not_in_ClinVar), In_ClinVar= mean(In_Clinvar)) %>% 
  mutate("% Uncertain" = round((Uncertain/Total),4)*100) %>% 
  select(vep_annotation, Total, In_ClinVar, "Pathogenic/Likely Pathogenic", "Benign/Likely Benign", Uncertain,"Not in Clinvar", "% in ClinVar", "% Uncertain") %>% 
  add_row(vep_annotation="stop_variant", Total=0,In_ClinVar=0, "Pathogenic/Likely Pathogenic"=0, "Benign/Likely Benign"=0, Uncertain=0,"Not in Clinvar"=0, "% in ClinVar"=0, "% Uncertain"=0) %>% 
  arrange(vep_annotation) %>% 
  add_row(vep_annotation="All",Total = sum(.$Total), In_ClinVar = sum(.$In_ClinVar), "Pathogenic/Likely Pathogenic" = sum(.$"Pathogenic/Likely Pathogenic"), "Benign/Likely Benign"=sum(.$"Benign/Likely Benign"), Uncertain=sum(.$Uncertain),"Not in Clinvar"=sum(.$"Not in Clinvar"), "% in ClinVar"=round((In_ClinVar/Total)*100,2), "% Uncertain"=round((Uncertain/In_ClinVar)*100,2))
```

AMR
```{r}
atm_latino_table <- atm_latino_unique() %>%
  group_by(vep_annotation) %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation)) %>% 
  count(clin_var_clinical_significance) %>% mutate(Total= sum(n)) %>% 
  mutate(Percent_in_ClinVar=round(prop.table(n),4)*100) %>% 
  mutate(Percent_in_ClinVar = ifelse(is.na(clin_var_clinical_significance), 0, Percent_in_ClinVar)) %>%
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>%
  replace_na(list("Pathogenic" = 0, "Uncertain" = 0, "Benign" = 0, "NA" = 0)) %>%
  rename(Not_in_ClinVar = "NA") %>%
  mutate(In_Clinvar = sum(Pathogenic, Benign, Uncertain)) %>% 
  summarise("% in ClinVar" = sum(Percent_in_ClinVar), Total=mean(Total), "Benign/Likely Benign"=sum(Benign), "Pathogenic/Likely Pathogenic"=sum(Pathogenic), Uncertain=sum(Uncertain), "Not in Clinvar"=sum(Not_in_ClinVar), In_ClinVar= mean(In_Clinvar)) %>% 
  mutate("% Uncertain" = round((Uncertain/Total),4)*100) %>% 
  select(vep_annotation, Total, In_ClinVar, "Pathogenic/Likely Pathogenic", "Benign/Likely Benign", Uncertain,"Not in Clinvar", "% in ClinVar", "% Uncertain") %>% 
  add_row(vep_annotation="start_lost", Total=0, In_ClinVar=0, "Pathogenic/Likely Pathogenic"=0, "Benign/Likely Benign"=0, Uncertain=0,"Not in Clinvar"=0, "% in ClinVar"=0, "% Uncertain"=0) %>% 
  arrange(vep_annotation) %>% 
  add_row(vep_annotation="All",Total = sum(.$Total), In_ClinVar = sum(.$In_ClinVar), "Pathogenic/Likely Pathogenic" = sum(.$"Pathogenic/Likely Pathogenic"), "Benign/Likely Benign"=sum(.$"Benign/Likely Benign"), Uncertain=sum(.$Uncertain),"Not in Clinvar"=sum(.$"Not in Clinvar"), "% in ClinVar"=round((In_ClinVar/Total)*100,2), "% Uncertain"=round((Uncertain/In_ClinVar)*100,2))
```

NFE
```{r}
atm_european_table <- atm_european_unique() %>% 
  group_by(vep_annotation) %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation)) %>% 
  count(clin_var_clinical_significance) %>% mutate(Total= sum(n)) %>% 
  mutate(Percent_in_ClinVar=round(prop.table(n),4)*100) %>% 
  mutate(Percent_in_ClinVar = ifelse(is.na(clin_var_clinical_significance), 0, Percent_in_ClinVar)) %>%
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>%
  replace_na(list("Pathogenic" = 0, "Uncertain" = 0, "Benign" = 0, "NA" = 0)) %>%
  rename(Not_in_ClinVar = "NA") %>%
  mutate(In_Clinvar = sum(Pathogenic, Benign, Uncertain)) %>% 
  summarise("% in ClinVar" = sum(Percent_in_ClinVar), Total=mean(Total), "Benign/Likely Benign"=sum(Benign), "Pathogenic/Likely Pathogenic"=sum(Pathogenic), Uncertain=sum(Uncertain), "Not in Clinvar"=sum(Not_in_ClinVar), In_ClinVar= mean(In_Clinvar)) %>% 
  mutate("% Uncertain" = round((Uncertain/Total),4)*100) %>% 
  select(vep_annotation, Total, In_ClinVar, "Pathogenic/Likely Pathogenic", "Benign/Likely Benign", Uncertain,"Not in Clinvar", "% in ClinVar", "% Uncertain") %>% 
  arrange(vep_annotation) %>% 
  add_row(vep_annotation="All",Total = sum(.$Total), In_ClinVar = sum(.$In_ClinVar), "Pathogenic/Likely Pathogenic" = sum(.$"Pathogenic/Likely Pathogenic"), "Benign/Likely Benign"=sum(.$"Benign/Likely Benign"), Uncertain=sum(.$Uncertain),"Not in Clinvar"=sum(.$"Not in Clinvar"), "% in ClinVar"=round((In_ClinVar/Total)*100,2), "% Uncertain"=round((Uncertain/In_ClinVar)*100,2))
```

EAS
```{r}
atm_east_asian_table <- atm_east_asian_unique() %>%
  group_by(vep_annotation) %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation)) %>% 
  count(clin_var_clinical_significance) %>% mutate(Total= sum(n)) %>% 
  mutate(Percent_in_ClinVar=round(prop.table(n),4)*100) %>% 
  mutate(Percent_in_ClinVar = ifelse(is.na(clin_var_clinical_significance), 0, Percent_in_ClinVar)) %>%
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>%
  mutate(Benign = ifelse(!"Benign" %in% colnames(.), 0, Benign)) %>%
  replace_na(list("Pathogenic" = 0, "Uncertain" = 0, "Benign" = 0, "NA" = 0)) %>%
  rename(Not_in_ClinVar = "NA") %>%
  mutate(In_Clinvar = sum(Pathogenic, Benign, Uncertain)) %>% 
  summarise("% in ClinVar" = sum(Percent_in_ClinVar), Total=mean(Total), "Benign/Likely Benign"=sum(Benign), "Pathogenic/Likely Pathogenic"=sum(Pathogenic), Uncertain=sum(Uncertain), "Not in Clinvar"=sum(Not_in_ClinVar), In_ClinVar= mean(In_Clinvar)) %>% 
  mutate("% Uncertain" = round((Uncertain/Total),4)*100) %>% 
  select(vep_annotation, Total, In_ClinVar, "Pathogenic/Likely Pathogenic", "Benign/Likely Benign", Uncertain,"Not in Clinvar", "% in ClinVar", "% Uncertain") %>% 
  add_row(vep_annotation="start_lost", Total=0,In_ClinVar=0, "Pathogenic/Likely Pathogenic"=0, "Benign/Likely Benign"=0, Uncertain=0,"Not in Clinvar"=0, "% in ClinVar"=0, "% Uncertain"=0)%>% 
  arrange(vep_annotation) %>% 
  add_row(vep_annotation="All",Total = sum(.$Total), In_ClinVar = sum(.$In_ClinVar), "Pathogenic/Likely Pathogenic" = sum(.$"Pathogenic/Likely Pathogenic"), "Benign/Likely Benign"=sum(.$"Benign/Likely Benign"), Uncertain=sum(.$Uncertain),"Not in Clinvar"=sum(.$"Not in Clinvar"), "% in ClinVar"=round((In_ClinVar/Total)*100,2), "% Uncertain"=round((Uncertain/In_ClinVar)*100,2))
```

SAS
```{r}
atm_south_asian_table <- atm_south_asian_unique() %>% 
  group_by(vep_annotation) %>% 
  mutate(clin_var_clinical_significance= case_when(clin_var_clinical_significance %in% Pathogenic ~ "Pathogenic", clin_var_clinical_significance %in% Uncertain ~ "Uncertain", clin_var_clinical_significance %in% Benign ~ "Benign", clin_var_clinical_significance=="not provided" ~ NA, TRUE~clin_var_clinical_significance)) %>% mutate(vep_annotation=case_when(vep_annotation %in% Splice ~ "splice_site_variant", vep_annotation %in% Inframe ~ "inframe_variant", vep_annotation%in%Stop ~ "stop_variant", TRUE ~ vep_annotation)) %>% 
  count(clin_var_clinical_significance) %>% mutate(Total= sum(n)) %>% 
  mutate(Percent_in_ClinVar=round(prop.table(n),4)*100) %>% 
  mutate(Percent_in_ClinVar = ifelse(is.na(clin_var_clinical_significance), 0, Percent_in_ClinVar)) %>%
  pivot_wider(names_from = clin_var_clinical_significance, values_from = n) %>%
  mutate(Benign=0) %>%
  replace_na(list("Pathogenic" = 0, "Uncertain" = 0, "Benign" = 0, "NA" = 0)) %>%
  rename(Not_in_ClinVar = "NA") %>%
  mutate(In_Clinvar = sum(Pathogenic, Benign, Uncertain)) %>% 
  summarise("% in ClinVar" = sum(Percent_in_ClinVar), Total=mean(Total), "Benign/Likely Benign"=sum(Benign), "Pathogenic/Likely Pathogenic"=sum(Pathogenic), Uncertain=sum(Uncertain), "Not in Clinvar"=sum(Not_in_ClinVar), In_ClinVar= mean(In_Clinvar)) %>% 
  mutate("% Uncertain" = round((Uncertain/Total),4)*100) %>% 
  select(vep_annotation, Total, In_ClinVar, "Pathogenic/Likely Pathogenic", "Benign/Likely Benign", Uncertain,"Not in Clinvar", "% in ClinVar", "% Uncertain") %>% 
  add_row(vep_annotation="start_lost", Total=0,In_ClinVar=0, "Pathogenic/Likely Pathogenic"=0, "Benign/Likely Benign"=0, Uncertain=0,"Not in Clinvar"=0, "% in ClinVar"=0, "% Uncertain"=0) %>% 
  arrange(vep_annotation) %>% 
  add_row(vep_annotation="All",Total = sum(.$Total), In_ClinVar = sum(.$In_ClinVar), "Pathogenic/Likely Pathogenic" = sum(.$"Pathogenic/Likely Pathogenic"), "Benign/Likely Benign"=sum(.$"Benign/Likely Benign"), Uncertain=sum(.$Uncertain),"Not in Clinvar"=sum(.$"Not in Clinvar"), "% in ClinVar"=round((In_ClinVar/Total)*100,2), "% Uncertain"=round((Uncertain/In_ClinVar)*100,2))
```

ATM comparison of variants between population
```{r}
atm %>% filter(allele_count_african_african_american!=0) %>% count(chromosome) -> Total_atm_african
atm_african_unique() %>% count(chromosome) %>% inner_join(Total_atm_african, by="chromosome") %>% mutate(n.z=n.y-n.x) %>% mutate(Population="AFR") %>% select(Population, n.z, n.y, n.x) -> atm_african_uni

atm %>% filter(allele_count_latino_admixed_american!=0) %>% count(chromosome) -> Total_atm_latino
atm_latino_unique() %>% count(chromosome) %>% inner_join(Total_atm_latino, by="chromosome") %>% mutate(n.z=n.y-n.x) %>% mutate(Population="AMR") %>% select(Population, n.z, n.y, n.x) -> atm_latino_uni

atm %>% filter(allele_count_european_non_finnish!=0) %>% count(chromosome) -> Total_atm_europe
atm_european_unique() %>% count(chromosome) %>% inner_join(Total_atm_europe, by="chromosome") %>% mutate(n.z=n.y-n.x) %>% mutate(Population="NFE") %>% select(Population, n.z, n.y, n.x) -> atm_europe_uni

atm %>% filter(allele_count_east_asian!=0) %>% count(chromosome) -> Total_atm_east_asia
atm_african_unique() %>% count(chromosome) %>% inner_join(Total_atm_east_asia, by="chromosome") %>% mutate(n.z=n.y-n.x) %>% mutate(Population="EAS") %>% select(Population, n.z, n.y, n.x) -> atm_east_asia_uni

atm %>% filter(allele_count_south_asian!=0) %>% count(chromosome) -> Total_atm_south_asia
atm_african_unique() %>% count(chromosome) %>% inner_join(Total_atm_south_asia, by="chromosome") %>% mutate(n.z=n.y-n.x) %>% mutate(Population="SAS") %>% select(Population, n.z, n.y, n.x) -> atm_south_asia_uni

full_join(atm_african_uni, atm_latino_uni) %>% full_join(atm_europe_uni) %>% full_join(atm_east_asia_uni) %>% full_join(atm_south_asia_uni) %>%
  rename(Total=n.y, Unique=n.x, Shared=n.z) %>% 
  select(-Total) %>% 
  pivot_longer(cols = c(Unique, Shared), names_to = "Characteristics", values_to = "Values") %>%
  ggplot(aes(x=Population, y=Values, fill=Characteristics)) + geom_col(position="dodge") + geom_text(aes(label = Values), position = position_dodge(1)) + labs(title= "Comparison of shared and unique variants between populations 
for the ATM gene", y="Number of variants") + xlim("NFE", "AFR", "AMR", "EAS", "SAS")
ggsave(here("output/ATM_pop.png"), width = 8, height = 6, dpi = 300)
```



